{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e4e1390e",
   "metadata": {},
   "source": [
    "Analysis for Beneficial Microbes presentation\n",
    "\n",
    "Still preliminary and needs tweaks\n",
    "\n",
    "## Reimport scrubbed qiime table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a5d5e63",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate qiime2-2022.8\n",
    "\n",
    "project=/Users/mallott/Dropbox/Projects/REACH/BM_presentation_analysis\n",
    "\n",
    "qiime tools import \\\n",
    "    --input-path ${project}/fecal/decontaminated_samples_22_23.biom \\\n",
    "    --type 'FeatureTable[Frequency]' \\\n",
    "    --input-format BIOMV100Format \\\n",
    "    --output-path ${project}/fecal/scrubbed_table.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "    --i-table ${project}/fecal/scrubbed_table.qza \\\n",
    "    --m-sample-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "    --o-visualization ${project}/fecal/scrubbed_table.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92c9411a",
   "metadata": {},
   "source": [
    "## Determine cutoffs for each sample type\n",
    "\n",
    "I split the fecal samples into their own table and the soil samples into their own table. Then I created visualizations for both classified tables, built phylogenetic trees, and ran alpha rarefaction on both to determine rarefaction cutoffs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a458861",
   "metadata": {},
   "outputs": [],
   "source": [
    "qiime feature-table filter-samples \\\n",
    "    --i-table ${project}/fecal/scrubbed_table.qza \\\n",
    "    --m-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "    --p-where \"[Type]='Feces'\" \\\n",
    "    --o-filtered-table ${project}/fecal/scrubbed_fecal_table.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --m-sample-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "    --o-visualization ${project}/fecal/scrubbed_fecal_table.qzv\n",
    "\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "    --i-sequences ${project}/fecal/rep-seqs-combined-1.qza \\\n",
    "    --o-alignment ${project}/fecal/aligned-rep-seqs-fecal.qza \\\n",
    "    --o-masked-alignment ${project}/fecal/masked-aligned-rep-seqs-fecal.qza \\\n",
    "    --o-tree ${project}/fecal/unrooted-tree-fecal.qza \\\n",
    "    --o-rooted-tree ${project}/fecal/rooted-tree-fecal.qza \\\n",
    "    --p-n-threads 3\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --i-phylogeny ${project}/fecal/rooted-tree-fecal.qza \\\n",
    "    --p-max-depth 10000 \\\n",
    "    --m-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "    --o-visualization ${project}/fecal/alpha-rarefaction-fecal.qzv\n",
    "\n",
    "qiime feature-table filter-samples \\\n",
    "    --i-table ${project}/fecal/scrubbed_table.qza \\\n",
    "    --m-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "    --p-where \"[Type]='Soil'\" \\\n",
    "    --o-filtered-table ${project}/soil/scrubbed_soil_table.qza\n",
    "\n",
    "qiime feature-table filter-samples \\\n",
    "    --i-table ${project}/fecal/scrubbed_soil_table.qza \\\n",
    "    --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "    --p-where \"[Site]='Illinois'\" \\\n",
    "    --o-filtered-table ${project}/soil/filtered_soil_table_il.qza\n",
    "\n",
    "qiime feature-table filter-samples \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il.qza \\\n",
    "    --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "    --p-where \"[Year]='2023'\" \\\n",
    "    --o-filtered-table ${project}/soil/filtered_soil_table_il_2023.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --m-sample-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "    --o-visualization ${project}/soil/filtered_soil_table_il_2023.qzv\n",
    "\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "    --i-sequences ${project}/fecal/rep-seqs-combined-1.qza \\\n",
    "    --o-alignment ${project}/soil/aligned-rep-seqs-soil.qza \\\n",
    "    --o-masked-alignment ${project}/soil/masked-aligned-rep-seqs-soil.qza \\\n",
    "    --o-tree ${project}/soil/unrooted-tree-soil.qza \\\n",
    "    --o-rooted-tree ${project}/soil/rooted-tree-soil.qza \\\n",
    "    --p-n-threads 6\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --i-phylogeny ${project}/soil/rooted-tree-soil.qza \\\n",
    "    --p-max-depth 20000 \\\n",
    "    --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "    --o-visualization ${project}/soil/alpha-rarefaction-soil-il2023.qzv\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --i-phylogeny ${project}/soil/rooted-tree-soil.qza \\\n",
    "    --p-max-depth 7500 \\\n",
    "    --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "    --o-visualization ${project}/soil/alpha-rarefaction-soil-il2023-7500.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88b6482f",
   "metadata": {},
   "source": [
    "#### Fecal sample cutoffs\n",
    "Shannon diversity levels off pretty quickly (>1000), but Faith's PD and observed features don't start leveling off until >4000. Rarefying to 1075 loses 1 sample from Illinois and 8 from Mississippi (total n after rarefying = 220). Rarefying to 4015 loses 15 from Illinois and 38 from Mississippi (total n after rarefying = 176). Probably need to run things both ways to be safe.\n",
    "\n",
    "#### Soil sample cutoffs\n",
    "3900 probably\n",
    "\n",
    "### Diversity Indices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55019d57",
   "metadata": {},
   "outputs": [],
   "source": [
    "qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny ${project}/fecal/rooted-tree-fecal.qza \\\n",
    "  --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "  --p-sampling-depth 4015 \\\n",
    "  --m-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "  --output-dir ${project}/fecal/core-metrics-results-4015\n",
    "\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny ${project}/fecal/rooted-tree-fecal.qza \\\n",
    "  --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "  --p-sampling-depth 1075 \\\n",
    "  --m-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "  --output-dir ${project}/fecal/core-metrics-results-1075\n",
    "\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny ${project}/soil/rooted-tree-soil.qza \\\n",
    "  --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "  --p-sampling-depth 3900 \\\n",
    "  --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "  --output-dir ${project}/soil/core-metrics-results-3900\n",
    "\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny ${project}/soil/rooted-tree-soil.qza \\\n",
    "  --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "  --p-sampling-depth 6050 \\\n",
    "  --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "  --output-dir ${project}/soil/core-metrics-results-6050"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03ba353a",
   "metadata": {},
   "source": [
    "### Taxonomic classification\n",
    "Then I used the Greengenes2 Naive Bayesian Classifer."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a960c9f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "qiime feature-classifier classify-sklearn \\\n",
    "    --i-classifier 2022.10.backbone.full-length.nb.qza \\\n",
    "    --i-reads ${project}/fecal/rep-seqs-combined-1.qza \\\n",
    "    --o-classification ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-n-jobs 4\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "  --m-input-file ${project}/fecal/taxonomy-fecal.qza \\\n",
    "  --o-visualization ${project}/fecal/taxonomy-fecal.qzv\n",
    "\n",
    "qiime taxa barplot \\\n",
    "  --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "  --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "  --m-metadata-file ${project}/fecal/metadata-combined.txt \\\n",
    "  --o-visualization ${project}/fecal/taxa-bar-plots-fecal.qzv\n",
    "\n",
    "qiime taxa barplot \\\n",
    "  --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "  --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "  --m-metadata-file ${project}/soil/Combined_16S_Metadata.txt \\\n",
    "  --o-visualization ${project}/soil/taxa-bar-plots-soil.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ebf83872",
   "metadata": {},
   "source": [
    "### Extract various things for analysis\n",
    "I then extracted alpha diversity indices, beta diversity indices, and feature tables for downstream analysis in R."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00ded5e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "qiime tools export --input-path ${project}/fecal/core-metrics-results-4015/shannon_vector.qza \\\n",
    "    --output-path ${project}/fecal/shannon\n",
    "qiime tools export --input-path ${project}/fecal/core-metrics-results-4015/evenness_vector.qza \\\n",
    "    --output-path ${project}/fecal/evenness\n",
    "qiime tools export --input-path ${project}/fecal/core-metrics-results-4015/faith_pd_vector.qza \\\n",
    "    --output-path ${project}/fecal/faith_pd\n",
    "qiime tools export --input-path ${project}/fecal/core-metrics-results-4015/observed_features_vector.qza \\\n",
    "    --output-path ${project}/fecal/observed\n",
    "cp ${project}/fecal/shannon/alpha-diversity.tsv ${project}/fecal/shannon-alpha-diversity.tsv\n",
    "cp ${project}/fecal/evenness/alpha-diversity.tsv ${project}/fecal/evenness-alpha-diversity.tsv\n",
    "cp ${project}/fecal/faith_pd/alpha-diversity.tsv ${project}/fecal/faithpd-alpha-diversity.tsv\n",
    "cp ${project}/fecal/observed/alpha-diversity.tsv ${project}/fecal/observed-alpha-diversity.tsv\n",
    "\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/core-metrics-results-4015/unweighted_unifrac_distance_matrix.qza \\\n",
    "    --output-path ${project}/fecal/core-metrics-results-4015/extracted_unweighted_unifrac\n",
    "cp ${project}/fecal/core-metrics-results-4015/extracted_unweighted_unifrac/distance-matrix.tsv \\\n",
    "    ${project}/fecal/unweighted-distance-matrix-4015.tsv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/core-metrics-results-4015/weighted_unifrac_distance_matrix.qza \\\n",
    "    --output-path ${project}/fecal/core-metrics-results-4015/extracted_weighted_unifrac\n",
    "cp ${project}/fecal/core-metrics-results-4015/extracted_weighted_unifrac/distance-matrix.tsv \\\n",
    "    ${project}/fecal/weighted-distance-matrix-4015.tsv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/core-metrics-results-4015/bray_curtis_distance_matrix.qza \\\n",
    "    --output-path ${project}/fecal/core-metrics-results-4015/extracted_bray_curtis\n",
    "cp ${project}/fecal/core-metrics-results-4015/extracted_bray_curtis/distance-matrix.tsv \\\n",
    "    ${project}/fecal/bray-distance-matrix-4015.tsv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/core-metrics-results-4015/jaccard_distance_matrix.qza \\\n",
    "    --output-path ${project}/fecal/core-metrics-results-4015/extracted_jaccard\n",
    "cp ${project}/fecal/core-metrics-results-4015/extracted_jaccard/distance-matrix.tsv \\\n",
    "    ${project}/fecal/jaccard-distance-matrix-4015.tsv\n",
    "\n",
    "qiime taxa collapse \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-level 2 --o-collapsed-table ${project}/fecal/scrubbed_fecal_table_phyla.qza\n",
    "qiime taxa collapse \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-level 5 --o-collapsed-table ${project}/fecal/scrubbed_fecal_table_family.qza\n",
    "qiime taxa collapse \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-level 6 --o-collapsed-table ${project}/fecal/scrubbed_fecal_table_genera.qza\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --output-path ${project}/fecal/table_asv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/scrubbed_fecal_table_phyla.qza \\\n",
    "    --output-path ${project}/fecal/table_phyla\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/scrubbed_fecal_table_family.qza \\\n",
    "    --output-path ${project}/fecal/table_family\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/scrubbed_fecal_table_genera.qza \\\n",
    "    --output-path ${project}/fecal/table_genera\n",
    "biom convert -i ${project}/fecal/table_asv/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-asv.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/table_phyla/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-phyla.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/table_family/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-family.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/table_genera/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-genera.tsv --to-tsv\n",
    "\n",
    "\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table.qza \\\n",
    "    --output-dir ${project}/fecal/ra_table_asv\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table_phyla.qza \\\n",
    "    --output-dir ${project}/fecal/ra_table_phyla\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table_family.qza \\\n",
    "    --output-dir ${project}/fecal/ra_table_family\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/fecal/scrubbed_fecal_table_genera.qza \\\n",
    "    --output-dir ${project}/fecal/ra_table_genera\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/ra_table_asv/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/fecal/ra_table_asv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/ra_table_phyla/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/fecal/ra_table_phyla\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/ra_table_family/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/fecal/ra_table_family\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/fecal/ra_table_genera/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/fecal/ra_table_genera\n",
    "biom convert -i ${project}/fecal/ra_table_asv/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-asv-relab.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/ra_table_phyla/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-phyla-relab.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/ra_table_family/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-family-relab.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/ra_table_genera/feature-table.biom \\\n",
    "    -o ${project}/fecal/feature-table-genera-relab.tsv --to-tsv\n",
    "\n",
    "####Soil\n",
    "qiime tools export --input-path ${project}/soil/core-metrics-results-3900/shannon_vector.qza \\\n",
    "    --output-path ${project}/soil/shannon\n",
    "qiime tools export --input-path ${project}/soil/core-metrics-results-3900/evenness_vector.qza \\\n",
    "    --output-path ${project}/soil/evenness\n",
    "qiime tools export --input-path ${project}/soil/core-metrics-results-3900/faith_pd_vector.qza \\\n",
    "    --output-path ${project}/soil/faith_pd\n",
    "qiime tools export --input-path ${project}/soil/core-metrics-results-3900/observed_features_vector.qza \\\n",
    "    --output-path ${project}/soil/observed\n",
    "cp ${project}/soil/shannon/alpha-diversity.tsv ${project}/soil/shannon-alpha-diversity.tsv\n",
    "cp ${project}/soil/evenness/alpha-diversity.tsv ${project}/soil/evenness-alpha-diversity.tsv\n",
    "cp ${project}/soil/faith_pd/alpha-diversity.tsv ${project}/soil/faithpd-alpha-diversity.tsv\n",
    "cp ${project}/soil/observed/alpha-diversity.tsv ${project}/soil/observed-alpha-diversity.tsv\n",
    "\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/core-metrics-results-3900/unweighted_unifrac_distance_matrix.qza \\\n",
    "    --output-path ${project}/soil/core-metrics-results-3900/extracted_unweighted_unifrac\n",
    "cp ${project}/soil/core-metrics-results-3900/extracted_unweighted_unifrac/distance-matrix.tsv \\\n",
    "    ${project}/soil/unweighted-distance-matrix-3900.tsv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/core-metrics-results-3900/weighted_unifrac_distance_matrix.qza \\\n",
    "    --output-path ${project}/soil/core-metrics-results-3900/extracted_weighted_unifrac\n",
    "cp ${project}/soil/core-metrics-results-3900/extracted_weighted_unifrac/distance-matrix.tsv \\\n",
    "    ${project}/soil/weighted-distance-matrix-3900.tsv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/core-metrics-results-3900/bray_curtis_distance_matrix.qza \\\n",
    "    --output-path ${project}/soil/core-metrics-results-3900/extracted_bray_curtis\n",
    "cp ${project}/soil/core-metrics-results-3900/extracted_bray_curtis/distance-matrix.tsv \\\n",
    "    ${project}/soil/bray-distance-matrix-3900.tsv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/core-metrics-results-3900/jaccard_distance_matrix.qza \\\n",
    "    --output-path ${project}/soil/core-metrics-results-3900/extracted_jaccard\n",
    "cp ${project}/soil/core-metrics-results-3900/extracted_jaccard/distance-matrix.tsv \\\n",
    "    ${project}/soil/jaccard-distance-matrix-3900.tsv\n",
    "\n",
    "qiime taxa collapse \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-level 2 --o-collapsed-table ${project}/soil/filtered_soil_table_il_2023_phyla.qza\n",
    "qiime taxa collapse \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-level 5 --o-collapsed-table ${project}/soil/filtered_soil_table_il_2023_family.qza\n",
    "qiime taxa collapse \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --i-taxonomy ${project}/fecal/taxonomy-fecal.qza \\\n",
    "    --p-level 6 --o-collapsed-table ${project}/soil/filtered_soil_table_il_2023_genera.qza\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --output-path ${project}/soil/table_asv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/filtered_soil_table_il_2023_phyla.qza \\\n",
    "    --output-path ${project}/soil/table_phyla\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/filtered_soil_table_il_2023_family.qza \\\n",
    "    --output-path ${project}/soil/table_family\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/filtered_soil_table_il_2023_genera.qza \\\n",
    "    --output-path ${project}/soil/table_genera\n",
    "biom convert -i ${project}/soil/table_asv/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-asv.tsv --to-tsv\n",
    "biom convert -i ${project}/soil/table_phyla/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-phyla.tsv --to-tsv\n",
    "biom convert -i ${project}/soil/table_family/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-family.tsv --to-tsv\n",
    "biom convert -i ${project}/soil/table_genera/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-genera.tsv --to-tsv\n",
    "\n",
    "\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023.qza \\\n",
    "    --output-dir ${project}/soil/ra_table_asv\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023_phyla.qza \\\n",
    "    --output-dir ${project}/soil/ra_table_phyla\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023_family.qza \\\n",
    "    --output-dir ${project}/soil/ra_table_family\n",
    "qiime feature-table relative-frequency \\\n",
    "    --i-table ${project}/soil/filtered_soil_table_il_2023_genera.qza \\\n",
    "    --output-dir ${project}/soil/ra_table_genera\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/ra_table_asv/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/soil/ra_table_asv\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/ra_table_phyla/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/soil/ra_table_phyla\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/ra_table_family/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/soil/ra_table_family\n",
    "qiime tools export \\\n",
    "    --input-path ${project}/soil/ra_table_genera/relative_frequency_table.qza \\\n",
    "    --output-path ${project}/soil/ra_table_genera\n",
    "biom convert -i ${project}/soil/ra_table_asv/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-asv-relab.tsv --to-tsv\n",
    "biom convert -i ${project}/soil/ra_table_phyla/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-phyla-relab.tsv --to-tsv\n",
    "biom convert -i ${project}/soil/ra_table_family/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-family-relab.tsv --to-tsv\n",
    "biom convert -i ${project}/fecal/ra_table_genera/feature-table.biom \\\n",
    "    -o ${project}/soil/feature-table-genera-relab.tsv --to-tsv\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
