---
title: "REACH_SDoH_Microbiome"
author: "Carlye Chaney"
date: "2024-09-23"
output: html_document
---

#Background
The gut microbiome has numerous impacts on human health, including gastrointestinal and cardiovascular disease risk. While social determinants of health (SDoH) are established predictors of the human gut microbiome, few researchers have assessed the direct and indirect effects of multiple SDoH in driving microbiome composition and diversity. In this project, we investigate the causal pathway linking SDoH to changes in the gut microbiome in two low-resource communities of the United States. This research will further our understanding of the ways in which structural and social factors contribute to the embodiment of inequality.

#Research Question
What is the causal pathway linking SDoH to changes in the gut microbiome in low-resource communities of the United States? 
    Aim 1: Quantify the association between SDoH variables and gut microbiome composition and diversity             in low resource communities.
                Aim 1a: Identify which SDoH variables have direct effects.
                Aim 1b: Identify which SDoH variables have indirect effects.
    Aim 2: Assess evidence for inflammation as a mechanism that links SDoH and the gut microbiome.


```{r}
#Get metadata into R for cleaning 
library(readxl)
REACH_metadata <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\REACH_2022&2023_METADATA_clean.xlsx")

#Checking to make sure the site columns are consistent within the data sheet
REACH_metadata$Site...95[REACH_metadata$Site...95 == "Mississippi"] <- "MS"
REACH_metadata$Site...95[REACH_metadata$Site...95 == "Illinois"] <- "IL"
all(REACH_metadata$Site...95 == REACH_metadata$Site...5)
  #True

#dropping extra Family ID column and extra site column. Renaming Site.
REACH_metadata <- REACH_metadata[, -c(1,95)]
names(REACH_metadata)[4]<-paste("Site")
```

#Data Cleaning

Food insecurity -- UPDATE TO DROP MIDDLE LEVEL?
```{r}
#Replacing NAs with 0 for frequency since that should be everyone who said no to "Skip meals"
REACH_metadata$skip_meals_freq[is.na(REACH_metadata$skip_meals_freq)] <- 0
REACH_metadata$skip_meals_other_freq[is.na(REACH_metadata$skip_meals_other_freq)] <- 0

#getting the higher value between how frequently the participant skips meals and how frequently others in the household skip meals
REACH_metadata$food_insecurity <- pmax(REACH_metadata$skip_meals_freq, REACH_metadata$skip_meals_other_freq)

REACH_metadata$food_insecurity <- as.factor(REACH_metadata$food_insecurity)

#Now the responses for "food insecurity" represent the frequency: 
#0 = does not skip meals
#1 = One-two months 
#2 = Three-six months 
#3 = Seven-nine months 
#4 = Ten-twelve months 

```

Healthcare access
```{r}
REACH_metadata$healthcare_access <- ifelse(REACH_metadata$medical_source == 1 | REACH_metadata$medical_source == 0, 1, NA)
REACH_metadata$healthcare_access <- ifelse(REACH_metadata$medical_source == 2 | REACH_metadata$medical_source == 3, 0, REACH_metadata$healthcare_access)

REACH_metadata$healthcare_access <- as.factor(REACH_metadata$healthcare_access)

#Recoded so 1 = primary care, 2 = limited access (ER, Urgent Care, or don't go)
```

Healthcare utilization 
```{r}
#need to add "insurance yes/no question to metadata sheet" but code would look like: 

REACH_metadata$healthcare_utilization <- ifelse(REACH_metadata$Insurance == 1 & REACH_metadata$regular_dr_apt == 1, 0, NA)
REACH_metadata$healthcare_utilization <- ifelse(REACH_metadata$Insurance == 1 & REACH_metadata$regular_dr_apt == 0, 1, REACH_metadata$healthcare_utilization)
REACH_metadata$healthcare_utilization <- ifelse(REACH_metadata$Insurance == 0 & REACH_metadata$regular_dr_apt == 0, 2, REACH_metadata$healthcare_utilization)

#This makes a 3-level variable where 0 = insurance and regular appointments, 1 = has insurance but doesn't go regularly, and 2 = does not have insurance and irregular healthcare appointments. 

table(REACH_metadata$healthcare_access, REACH_metadata$healthcare_utilization)

```
Unfortunately, the table from the code above shows that most people with insufficient healthcare access are listed as having insurance and going to the doctor at least once a year. This seems to suggest that people were replying the question about "regular doctor's appointments" as including visits to Urgent Care or the ER. So, we will use healthcare access and not healthcare utilization in the model. 

Employment
```{r}
#combining employment levels into "2"
REACH_metadata$Employment_limited <- ifelse(REACH_metadata$Employment_status == 3, 2, REACH_metadata$Employment_status)
#making retired category
REACH_metadata$Employment_limited <- ifelse(REACH_metadata$Employment_status == 5, 3, REACH_metadata$Employment_limited)
#making other category 
REACH_metadata$Employment_limited <- ifelse(REACH_metadata$Employment_status == 4 | REACH_metadata$Employment_status == 6 | REACH_metadata$Employment_status == 7, 5, REACH_metadata$Employment_limited)
#making unable to work category 
REACH_metadata$Employment_limited <- ifelse(REACH_metadata$Employment_status == 8, 4, REACH_metadata$Employment_limited)

REACH_metadata$Employment_limited <- as.factor(REACH_metadata$Employment_limited)

#levels: 0 = Full-time; 1 = Part-time; 2 = unemployed; 3 = retired; 4 = unable to work; 5 = other
```

Overcrowding -- Defined as more than 1.01 person per room (excluding bathrooms and kitchen, as per US census bureau)
```{r}
#getting number of rooms total. Telling to ignore rows that have NAs.
REACH_metadata$rooms_overcrowd <- rowSums(REACH_metadata[,c("Number_bedrooms", "Number_livingrooms", "Number_other_rooms")], na.rm=TRUE)

#getting number of people per room
REACH_metadata$overcrowding_numeric <- (REACH_metadata$number_people_in_home / REACH_metadata$rooms_overcrowd)

#making binary score of overcrowding by selecting =< as not overcrowded and >1.01 as overcrowded
REACH_metadata$overcrowding_score <- ifelse(REACH_metadata$overcrowding_numeric <= 1, 0, NA)
REACH_metadata$overcrowding_score <- ifelse(REACH_metadata$overcrowding_numeric > 1, 1, REACH_metadata$overcrowding_score)

REACH_metadata$overcrowding_score <- as.factor(REACH_metadata$overcrowding_score)
```

Diet PCA
```{r}
#all diet questions are on the same scale so we don't need to normalize.
REACH_metadata <- REACH_metadata[-c(1,4,17,28,50,78,113),] #getting rid of the repeat rows
rownames(REACH_metadata) <- REACH_metadata$SampleID

diet_data <- REACH_metadata[,c("Coffee", "Alcohol", "Candy", "Baked_goods", "Frozen_desserts", "Carbonated_drinks", "Fast_food", "Fried_food", "Red_meat", "Processed_meat", "Butter_margarine",     "Milk_products", "Eggs", "High_fat_dressings", "Processed_snacks", "cooking_oils")]

library(corrplot)
corrplot(cor(diet_data, , use="pairwise.complete.obs"),method = "ellipse")

pca_results <- prcomp(~ Coffee + Alcohol+Candy+Baked_goods+Frozen_desserts+Carbonated_drinks+Fast_food+Fried_food+Red_meat+Processed_meat+Butter_margarine+Milk_products+Eggs+High_fat_dressings+Processed_snacks+cooking_oils, data = REACH_metadata, scale = TRUE)

print(summary(pca_results),digits=2,loadings=pca_results$loadings,cutoff=0)

screeplot(pca_results,type="lines",col="red",lwd=2,pch=19,cex=1.2,main="Scree Plot of Diet Data")

#dropping coffee and alcohol
pca_results2 <- prcomp(~Candy+Baked_goods+Frozen_desserts+Carbonated_drinks+Fast_food+Fried_food+Red_meat+Processed_meat+Butter_margarine+Milk_products+Eggs+High_fat_dressings+Processed_snacks+cooking_oils, data = REACH_metadata, scale = TRUE)

print(summary(pca_results2),digits=2,loadings=pca_results$loadings,cutoff=0)

screeplot(pca_results2,type="lines",col="red",lwd=2,pch=19,cex=1.2,main="Scree Plot of Diet Data")


#Extracting the first 8 PCs and getting them into the metadata dataframe.
axes <- predict(pca_results, newdata = REACH_metadata)
REACH_metadata <- cbind(REACH_metadata, axes)
```
Other Diet Measures
```{r}
#Fatty food score
REACH_metadata$fatty <- REACH_metadata$Fast_food + REACH_metadata$Fried_food + REACH_metadata$Processed_meat + REACH_metadata$High_fat_dressings + REACH_metadata$Processed_snacks + REACH_metadata$cooking_oils + REACH_metadata$Red_meat

#Dairy food score
REACH_metadata$dairy <- REACH_metadata$Butter_margarine + REACH_metadata$Milk_products 

#Sugar food score
REACH_metadata$sugar <- REACH_metadata$Candy + REACH_metadata$Baked_goods + REACH_metadata$Frozen_desserts

```


Other variables 
```{r}
#Income. Combining levels 5 & 6 because they are very small.
REACH_metadata$Income_range <- ifelse(REACH_metadata$Income_range == 6, 5, REACH_metadata$Income_range)

#Year. Changing year to 1 & 2 factor instead of text.
REACH_metadata$Study_year <- ifelse(REACH_metadata$Year == "2022", 1, 2)
#121 = 1, 124 = 2. 

#Education
REACH_metadata$Education <- as.factor(REACH_metadata$Education)

#< HS = 0; HS = 1; College no degree = 2; Associate = 3; Bachelor's = 4, Master's = 5; Professional = 6; Doctorate  = 7)
#No professional or doctorate in dataset

#BMI
REACH_metadata$BMI <- (REACH_metadata$Weight/(REACH_metadata$Height*REACH_metadata$Height))*10000

cor(REACH_metadata$BMI, REACH_metadata$Bodyfat_Percentage, use="complete.obs")
#0.75 correlation. 

#Home ownership
REACH_metadata$Own_home <- as.factor(REACH_metadata$Own_home)

#Study year
REACH_metadata$Year <- as.factor(REACH_metadata$Year)

#antibiotics 
REACH_metadata$Abx_last_month <- as.factor(REACH_metadata$Abx_last_month)

#Sex
REACH_metadata$AQ4_Sex <- as.factor(REACH_metadata$AQ4_Sex)

#Smoking
REACH_metadata$smoke_household <- as.factor(REACH_metadata$smoke_household)
```

##Allostatic Load
Allostatic load is consists of the cumulative effect of chronic stress and life events on an individualâ€™s physiology and is calculated based on the proportion of biomarker values for which an individual is considered at risk (McEwen and Stellar, 1993).

The variables we include in our Allostatic Load index are systolic blood pressure, diastolic blood pressure, CRP, IgG, triglycerides, hemoglobin, white blood cell counts. Also include HDL and LDL? 

-- drop WBC because the data is not good?

- Blood pressure: Following the American Heart Association's guidelines, <129 & <80 is normal/elevated, and >130 | >80 is hypertension.
- CRP: >10 mg/L = high risk
- IgG: <600 mg/dL = high risk 
-Hemoglobin: <12 g/dL for adult females and <13 g/dL for adult males = high risk
-Triglycerides: >150 mg/dL = high risk
- HDL: < 60 mg/dL = high risk
- LDL: >100 mg/dL = high risk


```{r}
REACH_metadata$BP_risk <- ifelse(REACH_metadata$Systolic_blood_pressure <= 129 & REACH_metadata$Diastolic_blood_pressure <= 80, 0, 1) #0 = low risk, 1 = high risk
REACH_metadata$BP_risk <- ifelse(REACH_metadata$BP_med == 1, 1, REACH_metadata$BP_risk) #0 = low risk, 1 = high risk


REACH_metadata$CRP_risk <- ifelse(REACH_metadata$CRP > 10, 1, 0) #0 = low risk, 1 = high risk

#REACH_metadata$IgG_risk <- ifelse(REACH_metadata$IgG < 6.0, 1, 0) #0 = low risk, 1 = high risk

REACH_metadata$Hb_risk <- ifelse(REACH_metadata$Hemoglobin < 12 & REACH_metadata$AQ4_Sex == 2, 1, 0)
REACH_metadata$Hb_risk <- ifelse(REACH_metadata$Hemoglobin < 13 & REACH_metadata$AQ4_Sex == 1, 1, REACH_metadata$Hb_risk) #0= low risk, 1 = high risk

REACH_metadata$Triglycerides_risk <- ifelse(REACH_metadata$Triglycerides >= 150 | REACH_metadata$cholesterol_med ==1, 1, 0) #0 = low, 1=high

REACH_metadata$BMI_risk <- ifelse(REACH_metadata$BMI >= 30 | REACH_metadata$BMI < 18.5, 1, 0)

REACH_metadata$HDL_risk <- ifelse(REACH_metadata$Cholesterol_HDL <= 60 | REACH_metadata$cholesterol_med ==1, 1, 0)

REACH_metadata$LDL_risk <- ifelse(REACH_metadata$Cholesterol_LDL >= 100 | REACH_metadata$cholesterol_med ==1, 1, 0)

REACH_metadata$non_hdl <- REACH_metadata$Cholesterol_total - REACH_metadata$Cholesterol_HDL
REACH_metadata$non_hdl_risk <- ifelse(REACH_metadata$non_hdl > 130, 1, 0)

REACH_metadata$WBC_risk <- ifelse(REACH_metadata$WBC < 5 |REACH_metadata$WBC > 10 & REACH_metadata$AQ4_Sex == 1, 1, 0)
REACH_metadata$WBC_risk <- ifelse(REACH_metadata$WBC < 4.5 |REACH_metadata$WBC > 11 & REACH_metadata$AQ4_Sex == 2, 1, REACH_metadata$WBC_risk)

REACH_metadata$sum_risk <- REACH_metadata$BP_risk + REACH_metadata$Hb_risk + REACH_metadata$CRP_risk + REACH_metadata$Triglycerides_risk + REACH_metadata$non_hdl_risk + REACH_metadata$WBC_risk + REACH_metadata$BMI_risk

#excluded IgG to match NHANES but it might prove to be a useful immune measure to add in.
```

#NHANES Allostatic load test
Testing our AL index against the traditional AL index using NHANES data. Build an index using top / bottom quantile vs clinical cutoffs of our specific variables.

First get data into R
```{r}
library(devtools)

devtools::install_github("SilentSpringInstitute/RNHANES")


library(RNHANES)
library(plyr)
library(dplyr)

files <- nhanes_data_files()
biochem1718 <- nhanes_load_data("BIOPRO", "2017-2018") #Standard Biochemistry  dataset
glu1718 <- nhanes_load_data("GLU_J", "2017-2018") #glucose dataset, should use glucose from here
crp1718 <- nhanes_load_data("HSCRP_J", "2017-2018") #High-Sensitivity C-Reactive Protein 
blood1718 <- nhanes_load_data("CBC_J", "2017-2018") #Complete Blood Count with 5-Part Differential - Whole Blood
smoking1718 <- nhanes_load_data("SMQ_J", "2017-2018") #Smoking dataset - cigarette use
repro1718 <- nhanes_load_data("RHQ_J", "2017-2018") #reproductive health questionnaire dataset
BMI1718 <- nhanes_load_data("BMX_J", "2017-2018") #Body Measures dataset
BP1718 <- nhanes_load_data("BPX_J", "2017-2018") #Blood pressure dataset
HDL1718 <- nhanes_load_data("HDL_J", "2017-2018") #HDL
LDL1718 <- nhanes_load_data("TRIGLY_J", "2017-2018") #LDL
Tchol1718 <- nhanes_load_data("TCHOL_J", "2017-2018") #total cholesterol
food_security1718 <- nhanes_load_data("FSQ_J", "2017-2018") #food security
occupation1718 <- nhanes_load_data("OCQ_J", "2017-2018") #employment info 
dietquality1718 <- nhanes_load_data("DR1IFF_J", "2017-2018") #diet info
healthcare1718 <- nhanes_load_data("HIQ_J", "2017-2018") #healthcare coverage
housing1718 <- nhanes_load_data("HOQ_J", "2017-2018") #housing characteristics 


#Demographics
demo1718 <- nhanes_load_data("DEMO_J", "2017-2018")

#biochem1718  #Standard Biochemistry  dataset
biochem1718.vars <- biochem1718[,c("SEQN", "LBXSAL", "LBXSAPSI", "LBXSCR", "LBXSTB", "LBXSGTSI", "LBXSIR", "LBXSTR")]

#glu1718  #glucose dataset, should use glucose from here
glu1718.vars <- glu1718[,c("SEQN", "LBXGLU")]

#crp1718  #High-Sensitivity C-Reactive Protein 
crp1718.vars <- crp1718[,c("SEQN", "LBXHSCRP")]

#blood1718 #Complete Blood Count with 5-Part Differential - Whole Blood
blood1718.vars <- blood1718[,c("SEQN", "LBXLYPCT", "LBXMCVSI", "LBXRDW", "LBXWBCSI", "LBXHGB")]

#smoking1718 #Smoking dataset - cigarette use
smoking1718.vars <- smoking1718[,c("SEQN", "SMD030", "SMQ040", "SMQ050Q", "SMQ050U", "SMQ020")]

#repro1718  #reproductive health questionnaire dataset
repro1718.vars <- repro1718[,c("SEQN", "RHD143", "RHD043", "RHQ171", "RHQ031")]

#BMI1718 #Body Measures dataset
BMI1718.vars <- BMI1718[,c("SEQN", "BMXBMI")]

#demo1718 #Demographic variables
demo1718.vars <- demo1718[,c("SEQN","INDFMPIR", "RIAGENDR", "DMDEDUC2", "RIDRETH1", "RIDAGEYR")]

# BP1718 #Blood Pressure variables
BP1718.vars <- BP1718[,c("BPXSY1", "BPXSY2", "BPXSY3", "BPXDI1", "BPXDI2", "BPXDI3", "SEQN")]
BP1718.vars$systolic <- rowMeans(BP1718.vars[1:3], na.rm = TRUE) #systolic bp
BP1718.vars$diastolic <- rowMeans(BP1718.vars[4:6], na.rm = TRUE) #diastolic bp

#TCHOL_J #cholesterol 
Tchol1718 <- nhanes_load_data("TCHOL_J", "2017-2018")


df1718 <- join_all(list(biochem1718.vars, glu1718.vars, crp1718.vars, blood1718.vars, smoking1718.vars, repro1718.vars, BMI1718.vars, demo1718.vars, BP1718.vars, HDL1718, LDL1718, Tchol1718), by = "SEQN")
#n = 6401
```


Creating comparable allostatic load score to REACH Study 
```{r}
#Generating subset of people aged 18-84 with biomarker data

allostatic_subset <- subset(df1718,RIDAGEYR>17 & RIDAGEYR <= 84) #n = 5533. Subsetting so children & extremely elderly don't skew highest/lowest quantile calculations. 

allostatic_subset$BP_risk_REACH <- ifelse(allostatic_subset$systolic <= 129 & allostatic_subset$diastolic <= 80, 0, 1) #0 = low risk, 1 = high risk

allostatic_subset$CRP_risk_REACH <- ifelse(allostatic_subset$LBXHSCRP > 1, 1, 0) #0 = low risk, 1 = high risk

allostatic_subset$WBC_risk <- ifelse(allostatic_subset$LBXWBCSI < 5 |allostatic_subset$LBXWBCSI > 10 & allostatic_subset$RIAGENDR == 1, 1, 0)
allostatic_subset$WBC_risk <- ifelse(allostatic_subset$LBXWBCSI < 4.5 |allostatic_subset$LBXWBCSI > 11 & allostatic_subset$RIAGENDR == 2, 1, allostatic_subset$WBC_risk)

allostatic_subset$Hb_risk_REACH <- ifelse(allostatic_subset$LBXHGB < 12 & allostatic_subset$RIAGENDR == 2, 1, 0)
allostatic_subset$Hb_risk_REACH <- ifelse(allostatic_subset$LBXHGB < 13 & allostatic_subset$RIAGENDR == 1, 1, allostatic_subset$Hb_risk_REACH) #0= low risk, 1 = high risk

allostatic_subset$Triglycerides_risk_REACH <- ifelse(allostatic_subset$LBXSTR >= 150, 1, 0) #0 = low, 1=high

allostatic_subset$HDL_risk_REACH <- ifelse(allostatic_subset$LBDHDD <= 60, 1, 0)

allostatic_subset$LDL_risk_REACH <- ifelse(allostatic_subset$LBDLDL >= 100, 1, 0)

allostatic_subset$BMI_risk_REACH <- ifelse(allostatic_subset$BMXBMI >= 30 | allostatic_subset$BMXBMI < 18.5, 1, 0)

allostatic_subset$cholesterol_risk_REACH <- ifelse(allostatic_subset$LBXTC > 200, 1, 0)
allostatic_subset$non_hdl <- allostatic_subset$LBXTC - allostatic_subset$LBDHDD
allostatic_subset$non_hdl_risk <- ifelse(allostatic_subset$non_hdl >= 130, 1, 0)

allostatic_subset$sum_risk_REACH <- allostatic_subset$BP_risk_REACH + allostatic_subset$Hb_risk_REACH + allostatic_subset$CRP_risk_REACH + allostatic_subset$Triglycerides_risk_REACH + allostatic_subset$BMI_risk_REACH + allostatic_subset$WBC_risk + allostatic_subset$non_hdl_risk
#n=2,327

#removed allostatic_subset$HDL_risk_REACH + allostatic_subset$LDL_risk_REACH and total cholesterol to improve the correlation. Decided to use non-HDL as likely the most relevant biological variable even though the correlation with the traditional measure worsens since the plot based on sex and race fits better (see ggplot code below). 

allostatic_subset$sum_risk_REACH2 <- allostatic_subset$BP_risk_REACH + allostatic_subset$Hb_risk_REACH + allostatic_subset$CRP_risk_REACH + allostatic_subset$Triglycerides_risk_REACH + allostatic_subset$BMI_risk_REACH + allostatic_subset$HDL_risk_REACH + allostatic_subset$LDL_risk_REACH + allostatic_subset$WBC_risk

```


Calculating Allostatic Load based on highest/lowest quantile
```{r}
#Establishing high risk quantiles
Albumin_risk_NHANES <- quantile(allostatic_subset$LBXSAL, probs=c(0.25),na.rm=TRUE) 
Creatinine_risk_NHANES <- quantile(allostatic_subset$LBXSCR,probs=c(0.75),na.rm=TRUE)
Glucose_risk_NHANES <- quantile(allostatic_subset$LBXGLU ,probs=c(0.75),na.rm=TRUE)
CRP_risk_NHANES <- quantile(allostatic_subset$LBXHSCRP ,probs=c(0.75),na.rm=TRUE)
Lymphocyte_risk_NHANES <- quantile(allostatic_subset$LBXLYPCT ,probs=c(0.75),na.rm=TRUE)
MCV_risk_NHANES <- quantile(allostatic_subset$LBXMCVSI ,probs=c(0.75),na.rm=TRUE)
RDW_risk_NHANES <- quantile(allostatic_subset$LBXRDW ,probs=c(0.75),na.rm=TRUE)
Alkaline_risk_NHANES <- quantile(allostatic_subset$LBXSAPSI ,probs=c(0.75),na.rm=TRUE)
WBC_risk_NHANES <- quantile(allostatic_subset$LBXWBCSI ,probs=c(0.75),na.rm=TRUE)

#Scoring risk by biomarker
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_albumin = case_when(LBXSAL>Albumin_risk_NHANES[1]~ 0, 
                                                 LBXSAL<=Albumin_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_creatinine= case_when(LBXSCR<Creatinine_risk_NHANES[1]~0,
                                                    LBXSCR>=Creatinine_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_glucose= case_when(LBXGLU<Glucose_risk_NHANES[1]~0, 
                                                LBXGLU>=Glucose_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_CRP= case_when(LBXHSCRP<CRP_risk_NHANES[1]~0, 
                                            LBXHSCRP>=CRP_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_lymphocyte= case_when(LBXLYPCT<Lymphocyte_risk_NHANES[1]~0,
                                                   LBXLYPCT>=Lymphocyte_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_MCV= case_when(LBXMCVSI<MCV_risk_NHANES[1]~0,
                                             LBXMCVSI>=MCV_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_RDW= case_when(LBXRDW<RDW_risk_NHANES[1]~0,
                                            LBXRDW>=RDW_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_alkaline= case_when(LBXSAPSI<Alkaline_risk_NHANES[1]~0,
                                                 LBXSAPSI>=Alkaline_risk_NHANES[1]~1)) 
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_WBC= case_when(LBXWBCSI<WBC_risk_NHANES[1]~0,
                                            LBXWBCSI>=WBC_risk_NHANES[1]~1))

#Calculating allostatic load 
library(sjstats)
allostatic_subset <- dplyr::mutate(allostatic_subset, AL=(mean_n(allostatic_subset[, c("AL_albumin", "AL_creatinine", "AL_glucose", "AL_CRP", "AL_lymphocyte", "AL_MCV", "AL_RDW", "AL_alkaline", "AL_WBC")], 9, digits=4)))
#n = 2,474 with a complete score

allostatic_subset$sum_quantile <- allostatic_subset$AL_albumin + allostatic_subset$AL_creatinine + allostatic_subset$AL_glucose + allostatic_subset$AL_CRP + allostatic_subset$AL_lymphocyte+allostatic_subset$AL_MCV+allostatic_subset$AL_RDW+allostatic_subset$AL_alkaline+allostatic_subset$AL_WBC

cor(allostatic_subset$AL, allostatic_subset$sum_risk_REACH, use = "complete.obs")
#0.37
cor(allostatic_subset$AL, allostatic_subset$sum_risk_REACH2, use = "complete.obs")
#0.36

```
Comparing apples to apples by creating an NHANES quantile risk index on the same variables as the sum risk score:
```{r}
#BP risk 
#hemoglobin
#triglycerides
#BMI
Systolic_risk_NHANES <- quantile(allostatic_subset$systolic, probs=c(0.75),na.rm=TRUE) 
Diastolic_risk_NHANES <- quantile(allostatic_subset$diastolic,probs=c(0.75),na.rm=TRUE)
Triglycerides_risk_NHANES <- quantile(allostatic_subset$LBXSTR ,probs=c(0.75),na.rm=TRUE)
BMI_risk_NHANES <- quantile(allostatic_subset$BMXBMI ,probs=c(0.75),na.rm=TRUE)
HB_risk_NHANES <- quantile(allostatic_subset$LBXHGB ,probs=c(0.25),na.rm=TRUE)

allostatic_subset <- dplyr::mutate(allostatic_subset, AL_systolic= case_when(systolic<Systolic_risk_NHANES[1]~0, 
                                            systolic>=Systolic_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_diastolic= case_when(diastolic<Diastolic_risk_NHANES[1]~0,
                                                   diastolic>=Diastolic_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_triglycerides= case_when(LBXSTR<Triglycerides_risk_NHANES[1]~0,
                                                   LBXSTR>=Triglycerides_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_BMI= case_when(BMXBMI<BMI_risk_NHANES[1]~0,
                                                   BMXBMI>=BMI_risk_NHANES[1]~1))
allostatic_subset <- dplyr::mutate(allostatic_subset, AL_HB= case_when(LBXHGB<HB_risk_NHANES[1]~0,
                                                   LBXHGB>=HB_risk_NHANES[1]~1))

allostatic_subset <- dplyr::mutate(allostatic_subset, AL2=(mean_n(allostatic_subset[, c("AL_CRP", "AL_WBC", "AL_diastolic", "AL_systolic", "AL_triglycerides", "AL_BMI", "AL_HB")], 7, digits=4)))

cor(allostatic_subset$AL, allostatic_subset$AL2, use = "complete.obs")
#0.40
cor(allostatic_subset$sum_risk_REACH, allostatic_subset$AL2, use = "complete.obs")
#0.61
cor(allostatic_subset$sum_risk_REACH2, allostatic_subset$AL2, use = "complete.obs")
#0.55

#This seems to suggest that the difference in correlation is largely due to the different variables and not using top 75% as "high risk" because the correlation between the AL measures with 9 biomarkers and the one with 7 is only 0.4 when they are creating using the top 75% as risk. 
#However, variable selection alone doesn't explain everything -- the top 75% AL measure and the REACH measure are only 0.62 correlated.

```

Validating the performance of the conceptual model on NHANES data
```{r}
library(SASxport)

crp1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\HSCRP_J.xpt")
food_security1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\FSQ_J.xpt") #food security
occupation1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\OCQ_J.xpt") #employment info 
dietquality1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\DR1TOT_J.xpt") #diet info
healthcare1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\HIQ_J.xpt") #healthcare coverage
housing1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\HOQ_J.xpt") #housing characteristics 
demo1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\DEMO_J.xpt")
smoking1718_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\SMQFAM_J.xpt") 
BMI_SAS <- read.xport("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\NHANES files\\BMX_J.xpt") 
  
df1718_psem <- join_all(list(crp1718_SAS, food_security1718_SAS, occupation1718_SAS, dietquality1718_SAS, healthcare1718_SAS, housing1718_SAS, demo1718_SAS, smoking1718_SAS, BMI_SAS), by = "SEQN")
#n = 8704

#data cleaning
#food security combine two questions
df1718_psem$foodinsecurity1 <- ifelse(df1718_psem$FSD041 == "9", NA, df1718_psem$FSD041)
df1718_psem$foodinsecurity2 <- ifelse(df1718_psem$FSD052 == "9", NA, df1718_psem$FSD052)

df1718_psem$foodinsecurity <- ifelse(df1718_psem$foodinsecurity1 == "1" & !is.na(df1718_psem$foodinsecurity2), df1718_psem$foodinsecurity2, 0)

#income: make sure levels make sense
df1718_psem$income <- ifelse(df1718_psem$INDHHIN2 == "99", NA, df1718_psem$INDHHIN2)
df1718_psem$income <- ifelse(df1718_psem$INDHHIN2 == "77", NA, df1718_psem$income)

df1718_psem$income1 <- ifelse(df1718_psem$income == 1 | df1718_psem$income == 2, 1, NA)
df1718_psem$income1 <- ifelse(df1718_psem$income == 3 | df1718_psem$income == 4, 2, df1718_psem$income1)
df1718_psem$income1 <- ifelse(df1718_psem$income == 5 | df1718_psem$income == 6, 3, df1718_psem$income1)
df1718_psem$income1 <- ifelse(df1718_psem$income == 7 | df1718_psem$income == 8, 4, df1718_psem$income1)
df1718_psem$income1 <- ifelse(df1718_psem$income == 9 | df1718_psem$income == 10, 5, df1718_psem$income1)
df1718_psem$income1 <- ifelse(df1718_psem$income == 14, 6, df1718_psem$income1)
df1718_psem$income1 <- ifelse(df1718_psem$income == 15, 7, df1718_psem$income1)

#education
df1718_psem$education <- ifelse(df1718_psem$DMDEDUC2 == "7", NA, df1718_psem$DMDEDUC2)
df1718_psem$education <- ifelse(df1718_psem$DMDEDUC2 == "9", NA, df1718_psem$education)

#employment: bin to match reach
df1718_psem$employment <- ifelse(df1718_psem$OCQ180 < 35, 1, 0)
df1718_psem$employment1 <- ifelse(df1718_psem$OCQ380 == 1 | df1718_psem$OCQ380 == 2 | df1718_psem$OCQ380 == 7 | df1718_psem$OCQ380 == 77 | df1718_psem$OCQ380 == 99 | df1718_psem$OCQ380 == 5 , 1, NA)#everything we don't want
df1718_psem$employment1 <- ifelse(df1718_psem$OCQ380 == 3, 2, df1718_psem$employment1) #retired = 2
df1718_psem$employment1 <- ifelse(df1718_psem$OCQ380 == 4 | df1718_psem$OCQ380 == 6, 3, df1718_psem$employment1) #unable to work for health reasons

df1718_psem$employment2 <- ifelse(df1718_psem$employment1 == 2, 2, NA)
df1718_psem$employment2 <- ifelse(df1718_psem$employment1 == 3, 3, df1718_psem$employment2)


df1718_psem$employment3 <- paste(df1718_psem$employment2,df1718_psem$employment)

df1718_psem$employment3 <- gsub("2 NA", "2", df1718_psem$employment3)
df1718_psem$employment3 <- gsub("3 NA", "3", df1718_psem$employment3)
df1718_psem$employment3 <- gsub("NA 0", "0", df1718_psem$employment3)
df1718_psem$employment3 <- gsub("NA 1", "1", df1718_psem$employment3)
df1718_psem$employment3 <- gsub("NA NA", NA, df1718_psem$employment3)


df1718_psem$employment3 <- as.numeric(df1718_psem$employment3)

#run PCA on the nutrients SOS

rownames(df1718_psem) <- df1718_psem$SEQN
Diet_PCA <- na.omit(df1718_psem[,c(93:97, 101:136)])

scaled_pca <- prcomp(Diet_PCA, scale = TRUE)

print(summary(scaled_pca),digits=2,loadings=scaled_pca$loadings,cutoff=0)

screeplot(scaled_pca,type="lines",col="red",lwd=2,pch=19,cex=1.2,main="Scree Plot of Diet Data")

#Extracting the first 8 PCs and getting them into the metadata dataframe.
axes <- predict(scaled_pca, newdata = df1718_psem)
df1718_psem <- cbind(df1718_psem, axes)

#make overcrowding score
#number of rooms
df1718_psem$rooms <- ifelse(df1718_psem$HOD050 == "777", NA, df1718_psem$HOD050)
df1718_psem$rooms <- ifelse(df1718_psem$HOD050 == "999", NA, df1718_psem$rooms)

#getting number of people per room
df1718_psem$overcrowding_numeric <- (df1718_psem$DMDHHSIZ / df1718_psem$rooms)

#making binary score of overcrowding by selecting =< as not overcrowded and >1.01 as overcrowded
df1718_psem$overcrowding_score <- ifelse(df1718_psem$overcrowding_numeric <= 1, 0, NA)
df1718_psem$overcrowding_score <- ifelse(df1718_psem$overcrowding_numeric > 1, 1, df1718_psem$overcrowding_score)

df1718_psem$overcrowding_score <- as.numeric(df1718_psem$overcrowding_score)

#home ownership clean
df1718_psem$own_home <- ifelse(df1718_psem$HOQ065 == 3 | df1718_psem$HOQ065 == 7 | df1718_psem$HOQ065 == 9, NA, df1718_psem$HOQ065)

#healthcare access clean 
df1718_psem$healthcare <- ifelse(df1718_psem$HIQ011 == 7 | df1718_psem$HIQ011 == 9, NA, df1718_psem$HIQ011)

#smoking
df1718_psem$smoking <- ifelse(df1718_psem$SMD460 == 2 | df1718_psem$SMD460 == 3, 1, df1718_psem$SMD460)
df1718_psem$smoking <- ifelse(df1718_psem$smoking == 777 | df1718_psem$smoking == 999, 1, df1718_psem$smoking)

#we won't be accounting for survey weights, so these results are not necessarily representative of the population 

#apparently a bunch of variables have extra metadata attached to them, so this step will remove that. 
df1718_psem_complete <- na.omit(df1718_psem)
df1718_psem$RIAGENDR <- as.integer(df1718_psem$RIAGENDR)
df1718_psem$RIDAGEYR <- as.integer(df1718_psem$RIDAGEYR)
df1718_psem$BMXBMI <- as.integer(df1718_psem$BMXBMI)
df1718_psem$LBXHSCRP <- as.integer(df1718_psem$LBXHSCRP)

nhanes_model <- psem(
  lmer(employment3 ~ education + RIDAGEYR + (1|RIAGENDR), data = df1718_psem),
  lm(income1 ~ employment3 + RIDAGEYR, data = df1718_psem),
  lm(healthcare ~ employment3 + RIDAGEYR, data = df1718_psem),
  lm(foodinsecurity ~ income1 + RIDAGEYR, data = df1718_psem), 
  lm(own_home ~ income1 + RIDAGEYR, data = df1718_psem),
  lm(overcrowding_score ~ income1 + RIDAGEYR, data = df1718_psem),
  lmer(LBXHSCRP ~ PC1 + own_home+ overcrowding_score+ healthcare + RIDAGEYR + BMXBMI + (1|RIAGENDR) + (1|smoking), data = df1718_psem),
  lmer(PC1 ~ foodinsecurity + RIDAGEYR + (1|RIAGENDR) + BMXBMI, data = df1718_psem))

#AIC 137622.533. Chi-Squared = 57346.071 with P-value = 0 and on 31 degrees of freedom. Fisher's C = 1742.702 with P-value = 0 and on 62 degrees of freedom


#The tests of directed separation indicated we should add: 
#education to income, healthcare, food insecurity, homeownership, overcrowding, PC1, and CRP
#BMI to employment, income, food security, and home ownership
#employment to food security, overcrowding, CRP, and PC1
#income to healthcare access, CRP, and PC1
#healthcare access to homeownership and overcrowding
#food insecurity to home ownership, overcrowding, and CRP
#add homeownership to overcrowding
#overcrowding to PC1



nhanes_model_refined <- psem(
  lmer(employment3 ~ BMXBMI + education + RIDAGEYR + (1|RIAGENDR), data = df1718_psem),
  lm(income1 ~ BMXBMI + education + employment3 + RIDAGEYR, data = df1718_psem),
  lm(healthcare ~ income1 + education + employment3 + RIDAGEYR, data = df1718_psem),
  lm(foodinsecurity ~ employment3 + BMXBMI + education + income1 + RIDAGEYR, data = df1718_psem), 
  lm(own_home ~ healthcare + foodinsecurity + BMXBMI + education + income1 + RIDAGEYR, data = df1718_psem),
  lm(overcrowding_score ~ healthcare + own_home + foodinsecurity + employment3 + education + income1 + RIDAGEYR, data = df1718_psem),
  lmer(LBXHSCRP ~ foodinsecurity + income1 + employment3 + education + PC1 + own_home+ overcrowding_score+ healthcare + RIDAGEYR + BMXBMI + (1|RIAGENDR) + (1|smoking), data = df1718_psem),
  lmer(PC1 ~ overcrowding_score + income1 + employment3 + education + foodinsecurity + RIDAGEYR + (1|RIAGENDR) + BMXBMI, data = df1718_psem))

#AIC = 81635.556. Chi-Squared = 1309.094 with P-value = 0 and on 6 degrees of freedom. Fisher's C = 14.583 with P-value = 0.265 and on 12 degrees of freedom



```


Next, let's assess the performance of AL measures on variables with known relationships to AL: 

Education = DMDEDUC2 
Poverty to income ratio = INDFMPIR
Ethnicity = RIDRETH1

Smoking -- recode to never, past or current smoker based on variables: now smoke cigarettes (SMQ040) and age started smoking cigarettes (SMD030)

```{r}
#More data cleaning
allostatic_subset$DMDEDUC2[allostatic_subset$DMDEDUC2==7] <- NA #change refused to NA for education 
allostatic_subset$DMDEDUC2[allostatic_subset$DMDEDUC2==9] <- NA #change don't know to NA
allostatic_subset$DMDEDUC2 <- as.factor(allostatic_subset$DMDEDUC2) #change to factor
allostatic_subset$INDFMPIR[allostatic_subset$INDFMPIR==0] <- NA # change missing to NA 
allostatic_subset$INDFMPIR <- as.numeric(allostatic_subset$INDFMPIR) #make sure numeric
allostatic_subset$RIDRETH1 <- as.factor(allostatic_subset$RIDRETH1) #change to factor
allostatic_subset$RIAGENDR <- as.factor(allostatic_subset$RIAGENDR) #change to factor

#making smoking variable
allostatic_subset$smoking <- 2 #current smoker
allostatic_subset$smoking[allostatic_subset$SMQ020==2]<-0 #never smoker
allostatic_subset$smoking[allostatic_subset$SMQ040==3]<-1 #past smoker

#Test AL & Poverty to income ratio
summary(lm(allostatic_subset$sum_risk_REACH ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))
summary(lm(allostatic_subset$AL ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))
summary(lm(allostatic_subset$AL2 ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))
summary(lm(allostatic_subset$sum_risk_REACH2 ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))

#Seems like the REACH version explains less variation than AL, but the patterns perhaps fit more of what we would expect?

ggplot(allostatic_subset, aes(x = RIDRETH1, y = sum_risk_REACH , color = RIAGENDR, shape = RIAGENDR, size = 3))  + stat_summary(fun = "mean", geom = "point")  +
   labs(title = "Allostatic Load by Ethnicity and Sex") +
     theme_minimal() +
     theme(legend.title = element_blank())    

#Our REACH AL measure seems to follow established patterns with white women having the lowest AL and Black women the highest. 

ggplot(allostatic_subset, aes(x = RIDRETH1, y = AL , color = RIAGENDR, shape = RIAGENDR, size = 3))  + stat_summary(fun = "mean", geom = "point")  +
   labs(title = "Allostatic Load by Ethnicity and Sex") +
     theme_minimal() +
     theme(legend.title = element_blank()) 

#The 9-biomarker AL measure also fits that pattern to a degree, but it doesn't fit as well as the REACH AL version.

ggplot(allostatic_subset, aes(x = RIDRETH1, y = sum_risk_REACH2 , color = RIAGENDR, shape = RIAGENDR, size = 3))  + stat_summary(fun = "mean", geom = "point")  +
   labs(title = "Allostatic Load by Ethnicity and Sex") +
     theme_minimal() +
     theme(legend.title = element_blank())    

```
Now testing again with the 2005-2006 dataset to incorporate IgE and see if it's consistent across waves. 
#```{r}
library(RNHANES)
library(plyr)
library(dplyr)

files <- nhanes_data_files()
biochem0506 <- nhanes_load_data("BIOPRO_D", "2005-2006") #Standard Biochemistry  dataset
glu0506 <- nhanes_load_data("GLU_D", "2005-2006") #glucose dataset, should use glucose from here
crp0506 <- nhanes_load_data("CRP_D", "2005-2006") #C-Reactive Protein 
blood0506 <- nhanes_load_data("CBC_D", "2005-2006") #Complete Blood Count with 5-Part Differential - Whole Blood
smoking0506 <- nhanes_load_data("SMQ_D", "2005-2006") #Smoking dataset - cigarette use
repro0506 <- nhanes_load_data("RHQ_D", "2005-2006") #reproductive health questionnaire dataset
BMI0506 <- nhanes_load_data("BMX_D", "2005-2006") #Body Measures dataset
BP0506 <- nhanes_load_data("BPX_D", "2005-2006") #Blood pressure dataset
HDL0506 <- nhanes_load_data("HDL_D", "2005-2006") #HDL
LDL0506 <- nhanes_load_data("TRIGLY_D", "2005-2006") #LDL
Tchol0506 <- nhanes_load_data("TCHOL_D", "2005-2006") #total cholesterol
tIgE0506 <- nhanes_load_data("AL_IGE_D", "2005-2006") #total IgE

#creatinine needs to be correct. Standard creatinine (mg/dL) = -0.016 + 0.978 X (NHANES 05-06 uncalibrated serum creatinine, mg/dL)
biochem0506$LBXSCR <- (-0.016 + (0.978 * biochem0506$LBXSCR))

#Demographics
demo0506 <- nhanes_load_data("DEMO_D", "2005-2006")

#biochem0506  #Standard Biochemistry  dataset
biochem0506.vars <- biochem0506[,c("SEQN", "LBXSAL", "LBXSAPSI", "LBXSCR", "LBXSTB", "LBXSGTSI", "LBXSIR", "LBXSTR")]

#glu0506  #glucose dataset, should use glucose from here
glu0506.vars <- glu0506[,c("SEQN", "LBXGLU")]

#crp0506  #High-Sensitivity C-Reactive Protein 
crp0506.vars <- crp0506[,c("SEQN", "LBXCRP")]

#blood0506 #Complete Blood Count with 5-Part Differential - Whole Blood
blood0506.vars <- blood0506[,c("SEQN", "LBXLYPCT", "LBXMCVSI", "LBXRDW", "LBXWBCSI", "LBXHGB")]

#smoking0506 #Smoking dataset - cigarette use
smoking0506.vars <- smoking0506[,c("SEQN", "SMD030", "SMQ040", "SMQ050Q", "SMQ050U", "SMQ020")]

#repro0506  #reproductive health questionnaire dataset
repro0506.vars <- repro0506[,c("SEQN", "RHD143", "RHD042", "RHQ171", "RHQ031")]

#BMI0506 #Body Measures dataset
BMI0506.vars <- BMI0506[,c("SEQN", "BMXBMI")]

#demo0506 #Demographic variables
demo0506.vars <- demo0506[,c("SEQN","INDFMPIR", "RIAGENDR", "DMDEDUC2", "RIDRETH1", "RIDAGEYR")]

# BP0506 #Blood Pressure variables
BP0506.vars <- BP0506[,c("BPXSY1", "BPXSY2", "BPXSY3", "BPXDI1", "BPXDI2", "BPXDI3", "SEQN")]
BP0506.vars$systolic <- rowMeans(BP0506.vars[1:3], na.rm = TRUE) #systolic bp
BP0506.vars$diastolic <- rowMeans(BP0506.vars[4:6], na.rm = TRUE) #diastolic bp

tIgE0506.vars <- tIgE0506[,c("SEQN", "LBXIGE")]

df0506 <- join_all(list(biochem0506.vars, glu0506.vars, crp0506.vars, blood0506.vars, smoking0506.vars, repro0506.vars, BMI0506.vars, demo0506.vars, BP0506.vars, HDL0506, LDL0506, Tchol0506, tIgE0506.vars), by = "SEQN")
#n = 6401
```

Creating comparable allostatic load score to REACH Study 2005-2006
#```{r}
#Generating subset of people aged 18-84 with biomarker data
allostatic_subset_0506 <- subset(df0506, RIDAGEYR>17 & RIDAGEYR <= 84) #n = 5533. Subsetting so children & extremely elderly don't skew highest/lowest quantile calculations. 

allostatic_subset_0506$BP_risk_REACH <- ifelse(allostatic_subset_0506$systolic <= 129 & allostatic_subset_0506$diastolic <= 80, 0, 1) #0 = low risk, 1 = high risk


allostatic_subset_0506$CRP_risk_REACH <- ifelse(allostatic_subset_0506$LBXCRP > 1, 1, 0) #0 = low risk, 1 = high risk

allostatic_subset_0506$WBC_risk <- ifelse(allostatic_subset_0506$LBXWBCSI < 5 |allostatic_subset_0506$LBXWBCSI > 10 & allostatic_subset_0506$RIAGENDR == 1, 1, 0)
allostatic_subset_0506$WBC_risk <- ifelse(allostatic_subset_0506$LBXWBCSI < 4.5 |allostatic_subset_0506$LBXWBCSI > 11 & allostatic_subset_0506$RIAGENDR == 2, 1, allostatic_subset$WBC_risk)

allostatic_subset_0506$Hb_risk_REACH <- ifelse(allostatic_subset_0506$LBXHGB < 12 & allostatic_subset_0506$RIAGENDR == 2, 1, 0)
allostatic_subset_0506$Hb_risk_REACH <- ifelse(allostatic_subset_0506$LBXHGB < 13 & allostatic_subset_0506$RIAGENDR == 1, 1, allostatic_subset_0506$Hb_risk_REACH) #0= low risk, 1 = high risk

allostatic_subset_0506$Triglycerides_risk_REACH <- ifelse(allostatic_subset_0506$LBXSTR >= 150, 1, 0) #0 = low, 1=high

allostatic_subset_0506$HDL_risk_REACH <- ifelse(allostatic_subset_0506$LBDHDD <= 60, 1, 0)

allostatic_subset_0506$LDL_risk_REACH <- ifelse(allostatic_subset_0506$LBDLDL >= 100, 1, 0)

allostatic_subset_0506$BMI_risk_REACH <- ifelse(allostatic_subset_0506$BMXBMI >= 30 | allostatic_subset_0506$BMXBMI <= 18.5, 1, 0)

allostatic_subset_0506$cholesterol_risk_REACH <- ifelse(allostatic_subset_0506$LBXTC > 200, 1, 0)
allostatic_subset_0506$non_hdl <- allostatic_subset_0506$LBXTC - allostatic_subset_0506$LBDHDD
allostatic_subset_0506$non_hdl_risk <- ifelse(allostatic_subset_0506$non_hdl >= 130, 1, 0)

allostatic_subset_0506$IgE_risk <- ifelse(allostatic_subset_0506$LBXIGE > 1000, 1, 0)


allostatic_subset_0506$sum_risk_REACH <- allostatic_subset_0506$BP_risk_REACH + allostatic_subset_0506$Hb_risk_REACH + allostatic_subset_0506$CRP_risk_REACH + allostatic_subset_0506$Triglycerides_risk_REACH + allostatic_subset_0506$BMI_risk_REACH + allostatic_subset_0506$WBC_risk + allostatic_subset_0506$non_hdl_risk
#n=2,327

allostatic_subset_0506$sum_risk_REACH_IgE <- allostatic_subset_0506$BP_risk_REACH + allostatic_subset_0506$Hb_risk_REACH + allostatic_subset_0506$CRP_risk_REACH + allostatic_subset_0506$Triglycerides_risk_REACH + allostatic_subset_0506$BMI_risk_REACH + allostatic_subset_0506$WBC_risk + allostatic_subset_0506$non_hdl_risk + allostatic_subset_0506$IgE_risk

#removed allostatic_subset$HDL_risk_REACH + allostatic_subset$LDL_risk_REACH to improve the correlation 
allostatic_subset_0506$sum_risk_REACH2 <- allostatic_subset_0506$BP_risk_REACH + allostatic_subset_0506$Hb_risk_REACH + allostatic_subset_0506$CRP_risk_REACH + allostatic_subset_0506$Triglycerides_risk_REACH + allostatic_subset_0506$BMI_risk_REACH + allostatic_subset_0506$HDL_risk_REACH + allostatic_subset_0506$LDL_risk_REACH + allostatic_subset_0506$WBC_risk
```


Calculating Allostatic Load based on highest/lowest quantile for 2005-2006
#```{r}


#Establishing high risk quantiles
Albumin_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXSAL, probs=c(0.25),na.rm=TRUE) 
Creatinine_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXSCR,probs=c(0.75),na.rm=TRUE)
Glucose_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXGLU ,probs=c(0.75),na.rm=TRUE)
CRP_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXHSCRP ,probs=c(0.75),na.rm=TRUE)
Lymphocyte_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXLYPCT ,probs=c(0.75),na.rm=TRUE)
MCV_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXMCVSI ,probs=c(0.75),na.rm=TRUE)
RDW_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXRDW ,probs=c(0.75),na.rm=TRUE)
Alkaline_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXSAPSI ,probs=c(0.75),na.rm=TRUE)
WBC_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXWBCSI ,probs=c(0.75),na.rm=TRUE)



#Scoring risk by biomarker
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_albumin = case_when(LBXSAL>Albumin_risk_NHANES[1]~ 0, 
                                                 LBXSAL<=Albumin_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_creatinine= case_when(LBXSCR<Creatinine_risk_NHANES[1]~0,
                                                    LBXSCR>=Creatinine_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_glucose= case_when(LBXGLU<Glucose_risk_NHANES[1]~0, 
                                                LBXGLU>=Glucose_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_CRP= case_when(LBXCRP<CRP_risk_NHANES[1]~0, 
                                            LBXCRP>=CRP_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_lymphocyte= case_when(LBXLYPCT<Lymphocyte_risk_NHANES[1]~0,
                                                   LBXLYPCT>=Lymphocyte_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_MCV= case_when(LBXMCVSI<MCV_risk_NHANES[1]~0,
                                             LBXMCVSI>=MCV_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_RDW= case_when(LBXRDW<RDW_risk_NHANES[1]~0,
                                            LBXRDW>=RDW_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_alkaline= case_when(LBXSAPSI<Alkaline_risk_NHANES[1]~0,
                                                 LBXSAPSI>=Alkaline_risk_NHANES[1]~1)) 
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_WBC= case_when(LBXWBCSI<WBC_risk_NHANES[1]~0,
                                            LBXWBCSI>=WBC_risk_NHANES[1]~1))

#Calculating allostatic load 
library(sjstats)
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL=(mean_n(allostatic_subset_0506[, c("AL_albumin", "AL_creatinine", "AL_glucose", "AL_CRP", "AL_lymphocyte", "AL_MCV", "AL_RDW", "AL_alkaline", "AL_WBC")], 9, digits=4)))
#n = 2,474 with a complete score

allostatic_subset_0506$sum_quantile <- allostatic_subset_0506$AL_albumin + allostatic_subset_0506$AL_creatinine + allostatic_subset_0506$AL_glucose + allostatic_subset_0506$AL_CRP + allostatic_subset_0506$AL_lymphocyte+allostatic_subset_0506$AL_MCV+allostatic_subset_0506$AL_RDW+allostatic_subset_0506$AL_alkaline+allostatic_subset_0506$AL_WBC

cor(allostatic_subset_0506$AL, allostatic_subset_0506$sum_risk_REACH, use = "complete.obs")
#0.25

cor(allostatic_subset_0506$AL, allostatic_subset_0506$sum_risk_REACH_IgE, use = "complete.obs")
#0.25

```

Comparing these NHANES variables to percent-based risk using same variables as REACH 2005-2006
#```{r}

Systolic_risk_NHANES_0506 <- quantile(allostatic_subset_0506$systolic, probs=c(0.75),na.rm=TRUE) 
Diastolic_risk_NHANES_0506 <- quantile(allostatic_subset_0506$diastolic,probs=c(0.75),na.rm=TRUE)
Triglycerides_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXSTR ,probs=c(0.75),na.rm=TRUE)
BMI_risk_NHANES_0506 <- quantile(allostatic_subset_0506$BMXBMI ,probs=c(0.125,0.875),na.rm=TRUE) #making BMI two-tailed risk. 
HB_risk_NHANES_0506 <- quantile(allostatic_subset_0506$LBXHGB ,probs=c(0.25),na.rm=TRUE)


allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_systolic= case_when(systolic<Systolic_risk_NHANES[1]~0, 
                                            systolic>=Systolic_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_diastolic= case_when(diastolic<Diastolic_risk_NHANES[1]~0,
                                                   diastolic>=Diastolic_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_triglycerides= case_when(LBXSTR<Triglycerides_risk_NHANES[1]~0,
                                                   LBXSTR>=Triglycerides_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_BMI= case_when(BMXBMI<BMI_risk_NHANES[1]~0,
                                                   BMXBMI>=BMI_risk_NHANES[1]~1))
allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL_HB= case_when(LBXHGB<HB_risk_NHANES[1]~0,
                                                   LBXHGB>=HB_risk_NHANES[1]~1))

allostatic_subset_0506 <- dplyr::mutate(allostatic_subset_0506, AL2=(mean_n(allostatic_subset[, c("AL_CRP", "AL_WBC", "AL_diastolic", "AL_systolic", "AL_triglycerides", "AL_BMI", "AL_HB")], 7, digits=4)))

cor(allostatic_subset_0506$AL, allostatic_subset_0506$AL2, use = "complete.obs")
#0.40
cor(allostatic_subset_0506$sum_risk_REACH, allostatic_subset_0506$AL2, use = "complete.obs")
#0.61

#This seems to suggest that the difference in correlation is largely due to the different variables and not using top 75% as "high risk" because the correlation between the AL measures with 9 biomarkers and the one with 7 is only 0.4 when they are creating using the top 75% as risk. 
#However, variable selection alone doesn't explain everything -- the top 75% AL measure and the REACH measure are only 0.62 correlated.

```
Next, let's assess the performance of AL measures on variables with known relationships to AL for 2005-2006. 

Education = DMDEDUC2 
Poverty to income ratio = INDFMPIR
Ethnicity = RIDRETH1

Smoking -- recode to never, past or current smoker based on variables: now smoke cigarettes (SMQ040) and age started smoking cigarettes (SMD030)

#```{r}
#More data cleaning
allostatic_subset_0506$DMDEDUC2[allostatic_subset_0506$DMDEDUC2==7] <- NA #change refused to NA for education 
allostatic_subset_0506$DMDEDUC2[allostatic_subset_0506$DMDEDUC2==9] <- NA #change don't know to NA
allostatic_subset_0506$DMDEDUC2 <- as.factor(allostatic_subset_0506$DMDEDUC2) #change to factor
allostatic_subset_0506$INDFMPIR[allostatic_subset_0506$INDFMPIR==0] <- NA # change missing to NA 
allostatic_subset_0506$INDFMPIR <- as.numeric(allostatic_subset_0506$INDFMPIR) #make sure numeric
allostatic_subset_0506$RIDRETH1 <- as.factor(allostatic_subset_0506$RIDRETH1) #change to factor
allostatic_subset_0506$RIAGENDR <- as.factor(allostatic_subset_0506$RIAGENDR) #change to factor

#making smoking variable
allostatic_subset_0506$smoking <- 2 #current smoker
allostatic_subset_0506$smoking[allostatic_subset_0506$SMQ020==2]<-0 #never smoker
allostatic_subset_0506$smoking[allostatic_subset_0506$SMQ040==3]<-1 #past smoker

#Test AL & Poverty to income ratio
summary(lm(allostatic_subset$sum_risk_REACH ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))
summary(lm(allostatic_subset$AL ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))
summary(lm(allostatic_subset$AL2 ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))
summary(lm(allostatic_subset$sum_risk_REACH2 ~ allostatic_subset$INDFMPIR + allostatic_subset$RIAGENDR + allostatic_subset$RIDAGEYR + allostatic_subset$smoking + allostatic_subset$DMDEDUC2 + allostatic_subset$RIDRETH1))

#Seems like the REACH version explains less variation than AL, but the patterns perhaps fit more of what we would expect?

ggplot(allostatic_subset_0506, aes(x = RIDRETH1, y = sum_risk_REACH , color = RIAGENDR, shape = RIAGENDR, size = 3))  + stat_summary(fun = "mean", geom = "point")  +
   labs(title = "Allostatic Load by Ethnicity and Sex") +
     theme_minimal() +
     theme(legend.title = element_blank())    

#Our REACH AL measure seems to follow established patterns with white women having the lowest AL and Black women the highest. The 2005-2006 data follows the same patterns as the 2017-2018 data. 

ggplot(allostatic_subset_0506, aes(x = RIDRETH1, y = sum_risk_REACH_IgE , color = RIAGENDR, shape = RIAGENDR, size = 3))  + stat_summary(fun = "mean", geom = "point")  +
   labs(title = "Allostatic Load by Ethnicity and Sex") +
     theme_minimal() +
     theme(legend.title = element_blank())   

#Adding in IgE seems to fit the expected pattern much worse; this makes sense because IgE is not a general measure of immune function but specific to allergy. 

ggplot(allostatic_subset_0506, aes(x = RIDRETH1, y = AL , color = RIAGENDR, shape = RIAGENDR, size = 3))  + stat_summary(fun = "mean", geom = "point")  +
   labs(title = "Allostatic Load by Ethnicity and Sex") +
     theme_minimal() +
     theme(legend.title = element_blank()) 

#The 9-biomarker AL measure also fits that pattern to a degree, but it doesn't fit as well as the REACH AL version. White women should have relatively lower values and Black women should have higher values. 

```

Now getting the microbiome data into R. 
```{r}
library(lme4)
library(vegan)
library(pairwiseAdonis)
library(qiime2R)

bray_4015 = read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\bray_curtis_distance_matrix.qza")
jaccard_4015 = read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\jaccard_distance_matrix.qza")
unweighted_unifrac_4015 = read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\unweighted_unifrac_distance_matrix.qza")
weighted_unifrac_4015 = read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\weighted_unifrac_distance_matrix.qza")

faith_pd_4015 <- read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\faith_pd_vector.qza")
faith_pd_final <- as.data.frame(faith_pd_4015$data)
faith_pd_final$Sample_ID <- faith_pd_final$V1
#Faith_pd value variable name is V2 and I'm too tired to change it right now. 

shannon_4015 <- read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\shannon_vector.qza")
shannon_final <- as.data.frame(shannon_4015$data)
shannon_final$Sample_ID <- rownames(shannon_final)

evenness_4015 <- read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\evenness_vector.qza")
evenness_final <- as.data.frame(evenness_4015$data)
evenness_final$Sample_ID <- rownames(evenness_final)

observedfeatures_4015 <- read_qza("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\core-metrics-results-4015\\observed_features_vector.qza")
observedfeatures_final <- as.data.frame(observedfeatures_4015$data)
observedfeatures_final$Sample_ID <- rownames(observedfeatures_final)

REACH_metadata_1 <- merge(REACH_metadata, faith_pd_final, by = "Sample_ID")
REACH_metadata_2 <- merge(REACH_metadata_1, shannon_final, by = "Sample_ID")
REACH_metadata_3 <- merge(REACH_metadata_2, evenness_final, by = "Sample_ID")
REACH_metadata_4 <- merge(REACH_metadata_3, observedfeatures_final, by = "Sample_ID")
#n = 172
```

#Descriptive Statistics
```{r}

#removing all NAs for relevant variables

REACH_metadata_5 <- REACH_metadata_4[complete.cases(REACH_metadata_4[ , c('PC1', 'food_insecurity', 'CRP', 'Site', 'AQ4_Sex', 'Abx_last_month', 'BMI', 'smoke_household', 'Income_range', 'Own_home', 'overcrowding_score', 'healthcare_access', 'Employment_limited', 'Education', 'shannon_entropy')]),]

REACH_metada_AllostaticLoad <- REACH_metadata_4[complete.cases(REACH_metadata_4[ , c('PC1', 'food_insecurity', 'CRP', 'Site', 'AQ4_Sex', 'Abx_last_month', 'BMI', 'smoke_household', 'Income_range', 'Own_home', 'overcrowding_score', 'healthcare_access', 'Employment_status', 'Education', 'shannon_entropy', 'sum_risk')]),]

REACH_metadata_5$food_insecurity <- as.factor(REACH_metadata_5$food_insecurity)
REACH_metadata_5$AQ4_Sex <- as.numeric(REACH_metadata_5$AQ4_Sex) 
REACH_metadata_5$smoke_household <- as.numeric(REACH_metadata_5$smoke_household) 
REACH_metadata_5$Own_home <- as.numeric(REACH_metadata_5$Own_home) 
REACH_metadata_5$overcrowding_score <- as.numeric(REACH_metadata_5$overcrowding_score) 
REACH_metadata_5$healthcare_access <- as.factor(REACH_metadata_5$healthcare_access) 
REACH_metadata_5$Employment_limited <- as.numeric(REACH_metadata_5$Employment_limited)
REACH_metadata_5$Education <- as.numeric(REACH_metadata_5$Education)
REACH_metadata_5$Abx_last_month <- as.numeric(REACH_metadata_5$Abx_last_month)

REACH_metadata_5$food_insecurity <- ifelse(REACH_metadata_5$food_insecurity == 5, 4, REACH_metadata_5$food_insecurity)

mean(REACH_metadata_5$AQ3_Age)
sd(REACH_metadata_5$AQ3_Age)
mean(REACH_metadata_5$BMI)
sd(REACH_metadata_5$BMI)
mean(REACH_metadata_5$CRP)
sd(REACH_metadata_5$CRP)
mean(REACH_metada_AllostaticLoad$sum_risk)
sd(REACH_metada_AllostaticLoad$sum_risk)
mean(REACH_metadata_5$sugar)
sd(REACH_metadata_5$sugar)
mean(REACH_metadata_5$fatty)
sd(REACH_metadata_5$fatty)
mean(REACH_metadata_5$dairy)
sd(REACH_metadata_5$dairy)

IL_only <- REACH_metadata_5[REACH_metadata_5$Site== 'IL',]
MS_only <- REACH_metadata_5[REACH_metadata_5$Site == 'MS',]

mean(IL_only$AQ3_Age)
sd(IL_only$AQ3_Age)
mean(IL_only$BMI)
sd(IL_only$BMI)
mean(IL_only$CRP)
sd(IL_only$CRP)
mean(IL_only$sum_risk, na.rm=TRUE)
sd(IL_only$sum_risk, na.rm=TRUE)
mean(IL_only$dairy)
sd(IL_only$dairy)
mean(IL_only$fatty)
sd(IL_only$fatty)


mean(MS_only$AQ3_Age)
sd(MS_only$AQ3_Age)
mean(MS_only$BMI)
sd(MS_only$BMI)
mean(MS_only$CRP)
sd(MS_only$CRP)
mean(MS_only$sum_risk, na.rm=TRUE)
sd(MS_only$sum_risk, na.rm=TRUE)
mean(MS_only$dairy)
sd(MS_only$dairy)
mean(MS_only$fatty)
sd(MS_only$fatty)



table(REACH_metadata_5$Site, REACH_metadata_5$AQ4_Sex)
table(REACH_metadata_5$AQ4_Sex)

table(IL_only$AQ4_Sex)
table(MS_only$AQ4_Sex)

table(REACH_metadata_5$food_insecurity)
table(IL_only$food_insecurity)
table(MS_only$food_insecurity)

table(REACH_metadata_5$Education)
table(IL_only$Education)
table(MS_only$Education)

table(REACH_metadata_5$Employment_limited)
table(IL_only$Employment_limited)
table(MS_only$Employment_limited)

table(REACH_metadata_5$healthcare_access)
table(IL_only$healthcare_access)
table(MS_only$healthcare_access)

table(REACH_metadata_5$overcrowding_score)
table(IL_only$overcrowding_score)
table(MS_only$overcrowding_score)

table(REACH_metadata_5$Income_range)
table(IL_only$Income_range)
table(MS_only$Income_range)

table(REACH_metadata_5$Year)

table(REACH_metadata_5$AmericanIndian_AlaskaNative)
table(REACH_metadata_5$Asian)
table(REACH_metadata_5$Black_AfricanAmerican)
table(REACH_metadata_5$Hispanic_Latino)
table(REACH_metadata_5$White)
table(REACH_metadata_5$other_race)
table(REACH_metadata_5$NativeHawaiian_PacificIslander)

table(IL_only$Race_ethnicity)
table(MS_only$Race_ethnicity)

```


Piecewise SEM
```{r}

set.seed(1018)

#devtools::install_github("jslefche/piecewiseSEM@devel")
library(piecewiseSEM)


REACH_metadata_5$healthcare_access <- ifelse(REACH_metadata_5$healthcare_access == 1, 0, 1)
#combining the highest two levels of food insecurity because samples are small within each bin
REACH_metadata_5$food_insecurity2 <- ifelse(REACH_metadata_5$food_insecurity == 5, 4, REACH_metadata_5$food_insecurity)
REACH_metadata_5$food_insecurity3 <- seq(0, 5, length.out = 4)[REACH_metadata_5$food_insecurity2]

REACH_metada_AllostaticLoad$healthcare_access <- ifelse(REACH_metada_AllostaticLoad$healthcare_access == 1, 0, 1)
REACH_metada_AllostaticLoad$food_insecurity2 <- ifelse(REACH_metada_AllostaticLoad$food_insecurity == 5, 4, REACH_metada_AllostaticLoad$food_insecurity)
REACH_metada_AllostaticLoad$food_insecurity3 <- seq(0, 5, length.out = 4)[REACH_metada_AllostaticLoad$food_insecurity2]

 


#making the first Piecewise SEM for shannon diversity & CRP as measure of inflammation. Lme is not good at multiple random effects if they aren't nested, so I switched to lme4 package to use lmer and glmer. 
#shannon_model1 <- psem(
 # lme(Employment_limited ~ Education + AQ3_Age + AQ4_Sex, data = REACH_metadata_5, random = ~1 | Site),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  #lme(Income_range ~ Employment_limited + AQ3_Age + AQ4_Sex, data = REACH_metadata_5, random = ~1 | Site),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  #lme(healthcare_access ~ Employment_limited + AQ3_Age + AQ4_Sex, data = REACH_metadata_5, random = ~1 | Site),
  #lme(food_insecurity ~ Income_range + AQ3_Age + AQ4_Sex + smoke_household + BMI, data = REACH_metadata_5, random = ~1 | Site),
  #lme(Own_home ~ Income_range + AQ3_Age + AQ4_Sex, data = REACH_metadata_5, random = ~1 | Site),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  #lme(overcrowding_score ~ Income_range + AQ3_Age + AQ4_Sex, data = REACH_metadata_5, random = ~1 | Site),
  #lme(CRP ~ PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + Own_home+ overcrowding_score+ healthcare_access +AQ3_Age + AQ4_Sex + smoke_household + BMI, data = REACH_metadata_5, random = ~1 | Site),
  #lme(PC1 ~ food_insecurity + AQ3_Age + AQ4_Sex + smoke_household + BMI, data = REACH_metadata_5, random = ~1 | Site),
  #lme(shannon_entropy ~ PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + Employment_status + healthcare_access+ CRP + overcrowding_score + AQ3_Age + AQ4_Sex + smoke_household + BMI + Abx_last_month, data = REACH_metadata_5, random = ~1 | Site))

```

I found that various packages were interfering with eachother, and PSEM would only correctly if I kicked everything to R script. Below is the R Code to save the dataframe. It is followed by the code that I use in the R console to get our results. 

```{r}

library(writexl)

write_xlsx(REACH_metadata_5, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\REACH_metadata_5_22March2025.xlsx")
write_xlsx(REACH_metada_AllostaticLoad, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\REACH_metada_AllostaticLoad_22March2025.xlsx")


```

This is the code to use in the R script: 

```{r}
library(readxl)

REACH_metadata_5 <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\REACH_metadata_5_22March2025.xlsx")
REACH_metada_AllostaticLoad <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\REACH_metada_AllostaticLoad_22March2025.xlsx")


library(piecewiseSEM)
library(lme4)
library(semEff)


REACH_metadata_5_unique_samples <- REACH_metadata_5 %>% distinct(SampleID_Original, .keep_all = TRUE) # View the dimensions of the new data frame to confirm cat("Original dimensions:", dim(REACH_metadata_5_complete), "\n") cat("Dimensions after dropping repeats:", dim(REACH_metadata_5_unique_samples), "\n") #drop the second instance of the 5 repeated individuals
dim(REACH_metadata_5_unique_samples) #131 individuals


shannon_model2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = REACH_metadata_5),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = REACH_metadata_5),
  lmer(shannon_entropy ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original), data = REACH_metadata_5))

#I also tested a version of the model that included 8 PCs for diet, but the AIC was very high and the model didn't converge, so I will keep it to just PC1. 

summary(shannon_model2)
plot(shannon_model2)

evenness_model_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = REACH_metadata_5),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = REACH_metadata_5),
  lmer(pielou_evenness ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original)  + (1|Abx_last_month), data = REACH_metadata_5))

faithPD_model_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = REACH_metadata_5),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = REACH_metadata_5),
  lmer(V2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original)  + (1|Abx_last_month), data = REACH_metadata_5))

observed_model_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = REACH_metadata_5),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = REACH_metadata_5),
  lmer(observed_features ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|SampleID_Original)  + (1|Abx_last_month) + (1|AQ4_Sex) +(1|smoke_household), data = REACH_metadata_5))

shannon_model3 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI, data = REACH_metadata_5),
    lmer(dairy ~ food_insecurity2 + AQ3_Age + (1|smoke_household) + BMI + (1|Site), data = REACH_metadata_5),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metadata_5),
    lmer(shannon_entropy ~ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original), data = REACH_metadata_5))

#lmer(fatty ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = REACH_metadata_5),
#fatty was too correlated with sugar so I had to remove it. 

evenness_model_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI, data = REACH_metadata_5),
    lmer(dairy ~ food_insecurity2 + AQ3_Age + (1|smoke_household) + BMI + (1|Site), data = REACH_metadata_5),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metadata_5),
    lmer(pielou_evenness ~ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month), data = REACH_metadata_5))

faithPD_model_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI, data = REACH_metadata_5),
    lmer(dairy ~ food_insecurity2 + AQ3_Age + (1|smoke_household) + BMI + (1|Site), data = REACH_metadata_5),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metadata_5),
    lmer(V2 ~ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month), data = REACH_metadata_5))

observed_model_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metadata_5, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = REACH_metadata_5), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metadata_5),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = REACH_metadata_5),
  #I reduced the number of predictors since I don't think BMI, smoking, etc will affect this relationship
  lmer(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI, data = REACH_metadata_5),
    lmer(dairy ~ food_insecurity2 + AQ3_Age + (1|smoke_household) + BMI + (1|Site), data = REACH_metadata_5),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metadata_5),
    lmer(observed_features ~ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|SampleID_Original) + (1|AQ4_Sex) + (1|smoke_household), data = REACH_metadata_5))

```
#CRP notes
Because of 0 variance within the model, I removed the following random effects from the equations (PC1 models):
  Employment: Study year
  Income: Year, site, sex
  Healthcare access: site, sex
  food insecurity: sex, year
  own home: sample ID, year, site, sex
  Overcrowding: sample ID, site, sex
  CRP: year, site, smoking
  PC1: sample ID, site, study year
  Shannon: site, abx
  evenness: site
  faith's PD: site, smoking, sex
  observed: site, study year
  
In fatty/dairy/sugary models, they were the same as above except:
  CRP: Sample ID, smoking, site, study year, and smoking
  Fatty: Site, study year, sample ID
  Dairy: sex, study year, and sample ID
  Sugary: sex, study year, sample ID, smoking, site
  shannon: site, abx
  evenness: Site
  Faith PD: Site, smoking, sex
  observed: site, study year, abx last month
  
Now doing AL models. Remember sample size is smaller (n=92)
```{r}
REACH_metada_AllostaticLoad$Employment_limited <- as.factor(REACH_metada_AllostaticLoad$Employment_limited)
REACH_metada_AllostaticLoad$overcrowding_score <- as.factor(REACH_metada_AllostaticLoad$overcrowding_score)
REACH_metada_AllostaticLoad$Own_home <- as.factor(REACH_metada_AllostaticLoad$Own_home)
REACH_metada_AllostaticLoad$Education <- as.factor(REACH_metada_AllostaticLoad$Education)

REACH_metada_AllostaticLoad$AQ3_Age_scaled <- seq(1, 6, length.out = 79)[REACH_metada_AllostaticLoad$AQ3_Age]

shannon_AL_1 <- psem(
  lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(shannon_entropy ~ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original), data = REACH_metada_AllostaticLoad))

evenness_AL_1 <- psem(
   lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(pielou_evenness ~ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original), data = REACH_metada_AllostaticLoad))
  
  

faithPD_AL_1 <- psem(
   lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(V2 ~ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original), data = REACH_metada_AllostaticLoad))
  

observed_AL_1 <- psem(
  lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lmer(observed_features ~ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI  + (1|SampleID_Original), data = REACH_metada_AllostaticLoad))
  

shannon_AL_2 <- psem(
  lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = REACH_metada_AllostaticLoad),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metada_AllostaticLoad),
  lmer(shannon_entropy ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|smoke_household) , data = REACH_metada_AllostaticLoad))
  


evenness_AL_2 <- psem(
  lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = REACH_metada_AllostaticLoad),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metada_AllostaticLoad),
  lmer(pielou_evenness ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|smoke_household) , data = REACH_metada_AllostaticLoad))


FaithPD_AL_2 <- psem(
   lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = REACH_metada_AllostaticLoad),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metada_AllostaticLoad),
  lmer(V2 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site) , data = REACH_metada_AllostaticLoad))
  


observed_AL_2 <- psem(
 lm(Employment_limited ~ Education + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = REACH_metada_AllostaticLoad),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = REACH_metada_AllostaticLoad, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = REACH_metada_AllostaticLoad), 
  lm(Own_home ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = REACH_metada_AllostaticLoad),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original), data = REACH_metada_AllostaticLoad),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = REACH_metada_AllostaticLoad),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = REACH_metada_AllostaticLoad),
  lmer(observed_features ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI  + (1|SampleID_Original) + (1|Site), data = REACH_metada_AllostaticLoad))

```
  
  sum_risk: site, sex, smoke
  shannon: abx, site, smoke, sex
  employment: Sex and study year
  income: sex, site, study year, ID
  healthcare access: ID, site and sex
  food insecurity: site, year, sex, and age, and ID *** had to take ID out for model to work
  own home: site, year, sex, and age
  overcrowding: site, (year, sex too?)
  PC1: Study year and site
  evenness: site, sex, abx last month
  Faith PD: site and smoke
  observed: smoke, site, sex, year

For the ones with dairy and sugar:
  sum risk: site, sex, smoking
  dairy: all
  sugar: all
  shannon: sex and abx
  evenness: sex, abx
  Faith's PD: sex, site, smoke
  observed: sex, site, study year

#Refining the alpha diversity models  
The Fisher's C indicated a poor fit for many of the alpha diversity models, and the tests of direct separation suggested that we were missing important relationships in the data. Based on the tests of direct separation, here are the refined models: 
```{r}
shannon_CRP_V2 <- psem(
  lm(Employment_limited ~ Abx_last_month+ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples),
  lm(shannon_entropy ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = REACH_metadata_5_unique_samples) )

shannonCRP_sem_effects <- semEff(shannon_CRP_V2, R = 1000, seed = 123)



evenness_CRP_V2 <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples),
  lm(pielou_evenness ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
  
evenness_CRP_sem_effects <- semEff(evenness_CRP_V2, R = 1000, seed = 123) 


FaithPD_CRP_V2 <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples),
  lm(V2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
  
FaithPD_CRP_sem_effects <- semEff(FaithPD_CRP_V2, R = 1000, seed = 123)


observed_CRP_V2 <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples),
  lm(observed_features ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
  
observed_CRP_V2_sem_effects <- semEff(observed_CRP_V2, R = 1000, seed = 123)

#Now comparing to the dairy and sugar frequency scores. SD = sugar & dairy 

shannon_CRP_SD <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples),
  lm(shannon_entropy ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
  
shannon_CRP_SD_sem_effects <- semEff(shannon_CRP_SD, R = 1000, seed = 123)
  
  
evenness_CRP_SD <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples),
  lm(pielou_evenness ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
    
evenness_CRP_SD_sem_effects <- semEff(evenness_CRP_SD, R = 1000, seed = 123)

FaithPD_CRP_SD <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples),
    lm(V2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
    
FaithPD_CRP_SD_sem_effects <- semEff(FaithPD_CRP_SD, R = 1000, seed = 123)

observed_CRP_SD <- psem(
  lm(Employment_limited ~ Abx_last_month+Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = REACH_metadata_5_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples),
  lm(observed_features ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples))
  
observed_CRP_SD_sem_effects <- semEff(observed_CRP_SD, R = 1000, seed = 123)

```

Now refined models for AL: 

```{r}
REACH_metada_AllostaticLoad$Employment_limited <- as.numeric(REACH_metada_AllostaticLoad$Employment_limited)
REACH_metada_AllostaticLoad$overcrowding_score <- as.numeric(REACH_metada_AllostaticLoad$overcrowding_score)
REACH_metada_AllostaticLoad$Own_home <- as.numeric(REACH_metada_AllostaticLoad$Own_home)
REACH_metada_AllostaticLoad$Education <- as.numeric(REACH_metada_AllostaticLoad$Education)
REACH_metada_AllostaticLoad$AQ4_Sex <- as.factor(REACH_metada_AllostaticLoad$AQ4_Sex)
REACH_metada_AllostaticLoad$Site <- as.factor(REACH_metada_AllostaticLoad$Site)
REACH_metada_AllostaticLoad$smoke_household <- as.factor(REACH_metada_AllostaticLoad$smoke_household)

REACH_metadata_5_unique_samples_AL <- REACH_metada_AllostaticLoad %>% distinct(SampleID_Original, .keep_all = TRUE) #drops two individuals, keeps first observation

shannon_AL_PC1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month + PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(c ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
shannon_AL_PC1_sem_effects <- semEff(shannon_AL_PC1, R = 1000, seed = 123)

evenness_AL_PC1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(pielou_evenness ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
evenness_AL_PC1_sem_effects <- semEff(evenness_AL_PC1, R = 1000, seed = 123)

FaithPD_AL_PC1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(V2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
FaithPD_AL_PC1_sem_effects <- semEff(FaithPD_AL_PC1, R = 1000, seed = 123)

observed_AL_PC1 <- psem(
 lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(observed_features ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
observed_AL_PC1_sem_effects <- semEff(observed_AL_PC1, R = 1000, seed = 123)

#Now dairy and sugar models 

shannon_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(shannon_entropy ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
shannon_AL_DS_sem_effects <- semEff(shannon_AL_DS, R = 1000, seed = 123)

evenness_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(pielou_evenness ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
evenness_AL_DS_sem_effects <- semEff(evenness_AL_DS, R = 1000, seed = 123)
  
FaithPD_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(V2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
FaithPD_AL_DS_sem_effects <- semEff(FaithPD_AL_DS, R = 1000, seed = 123)


observed_AL_DS <- psem(
 lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = REACH_metadata_5_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = REACH_metadata_5_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = REACH_metadata_5_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = REACH_metadata_5_unique_samples_AL),
  lm(sum_risk ~ Abx_last_month +dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household, data = REACH_metadata_5_unique_samples_AL),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = REACH_metadata_5_unique_samples_AL),
  lm(observed_features ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = REACH_metadata_5_unique_samples_AL))
  
observed_AL_DS_sem_effects <- semEff(observed_AL_DS, R = 1000, seed = 123)

```

Last thing is to check if "everything" model performs better than looking at a causal pathway
```{r}

#CRP_PC1
shannon_CRP_PC1_all <- psem(lm(shannon_entropy ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples))

evenness_CRP_PC1_all <- psem(lm(pielou_evenness ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples))

FaithPD_CRP_PC1_all <- psem(lm(V2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5))

observed_CRP_PC1_all <- psem(lm(observed_features ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples))

#Now dairy and sugar models for CRP

shannon_CRP_DS_all <- psem(lm(shannon_entropy ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+  healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + dairy + sugar, data = REACH_metadata_5_unique_samples))

evenness_CRP_DS_all <- psem(lm(pielou_evenness ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+  healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + dairy + sugar, data = REACH_metadata_5_unique_samples))

FaithPD_CRP_DS_all <- psem(lm(V2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + dairy + sugar, data = REACH_metadata_5_unique_samples))

observed_CRP_DS_all <- psem(lm(observed_features ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + dairy + sugar, data = REACH_metadata_5_unique_samples))

#AL_PC1

#(1|Study_year) + (1|SampleID_Original) + (1|smoke_household) + (1|AQ4_Sex) + (1|Site) + (1|Abx_last_month)

shannon_AL_PC1_all <- psem(lm(shannon_entropy ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples_AL))

evenness_AL_PC1_all <- psem(lm(pielou_evenness ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples_AL))

FaithPD_AL_PC1_all <- psem(lm(V2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples_AL))

observed_AL_PC1_all <- psem(lm(observed_features ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = REACH_metadata_5_unique_samples_AL))

#AL_DS

shannon_AL_DS_all <- psem(lm(shannon_entropy ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + sugar + dairy, data = REACH_metadata_5_unique_samples_AL))

evenness_AL_DS_all <- psem(lm(pielou_evenness ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + sugar + dairy, data = REACH_metadata_5_unique_samples_AL))

FaithPD_AL_DS_all <- psem(lm(V2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + sugar + dairy, data = REACH_metadata_5_unique_samples_AL))

observed_AL_DS_all <- psem(lm(observed_features ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month + sugar + dairy, data = REACH_metadata_5_unique_samples_AL))

```


##Beta Diversity##

First I need to calculate NMDS Scores and merge them with the metadata. 
```{r}
library(vegan)

mds_bray <-metaMDS(bray_4015$data, k=3, trymax=1000)
mds_bray<-mds_bray$points
mds_bray<-merge(x=mds_bray, y = REACH_metadata_5, 
                                  by.x = "row.names", by.y = "Sample_ID")

mds_jaccard <-metaMDS(jaccard_4015$data, k=3, trymax=1000)
mds_jaccard<-mds_jaccard$points
mds_jaccard<-merge(x=mds_jaccard, y = REACH_metadata_5, 
                                  by.x = "row.names", by.y = "Sample_ID")

mds_unweighted_unifrac <-metaMDS(unweighted_unifrac_4015$data, k=3, trymax=1000)
mds_unweighted_unifrac<-mds_unweighted_unifrac$points
mds_unweighted_unifrac<-merge(x=mds_unweighted_unifrac, y = REACH_metadata_5, 
                                  by.x = "row.names", by.y = "Sample_ID")


mds_weighted_unifrac <-metaMDS(weighted_unifrac_4015$data, k=2, trymax=1000)
mds_weighted_unifrac<-mds_weighted_unifrac$points
mds_weighted_unifrac<-merge(x=mds_weighted_unifrac, y = REACH_metadata_5, 
                                  by.x = "row.names", by.y = "Sample_ID")

library(writexl)

write_xlsx(mds_bray, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_bray_22April2025.xlsx")
write_xlsx(mds_jaccard, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_jaccard_22April2025.xlsx")
write_xlsx(mds_unweighted_unifrac, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_unweighted_unifrac_22April2025.xlsx")
write_xlsx(mds_weighted_unifrac, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_weighted_unifrac_22April2025.xlsx")


#REACH_metada_AllostaticLoad <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\REACH_metada_AllostaticLoad.xlsx")

mds_bray_AL <-metaMDS(bray_4015$data, k=3, trymax=1000)
mds_bray_AL <-mds_bray_AL$points
mds_bray_AL <- merge(x=mds_bray_AL, y = REACH_metada_AllostaticLoad, 
                                  by.x = "row.names", by.y = "Sample_ID")

mds_jaccard_AL <-metaMDS(jaccard_4015$data, k=3, trymax=1000)
mds_jaccard_AL<-mds_jaccard_AL$points
mds_jaccard_AL<-merge(x=mds_jaccard_AL, y = REACH_metada_AllostaticLoad, 
                                  by.x = "row.names", by.y = "Sample_ID")

mds_unweighted_unifrac_AL <-metaMDS(unweighted_unifrac_4015$data, k=3, trymax=1000)
mds_unweighted_unifrac_AL<-mds_unweighted_unifrac_AL$points
mds_unweighted_unifrac_AL<-merge(x=mds_unweighted_unifrac_AL, y = REACH_metada_AllostaticLoad, 
                                  by.x = "row.names", by.y = "Sample_ID")


mds_weighted_unifrac_AL <-metaMDS(weighted_unifrac_4015$data, k=2, trymax=1000)
mds_weighted_unifrac_AL<-mds_weighted_unifrac_AL$points
mds_weighted_unifrac_AL<-merge(x=mds_weighted_unifrac_AL, y = REACH_metada_AllostaticLoad, 
                                  by.x = "row.names", by.y = "Sample_ID")

library(writexl)

write_xlsx(mds_bray_AL, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_bray_AL_22April2025.xlsx")
write_xlsx(mds_jaccard_AL, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_jaccard_AL_22April2025.xlsx")
write_xlsx(mds_unweighted_unifrac_AL, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_unweighted_unifrac_AL_22April2025.xlsx")
write_xlsx(mds_weighted_unifrac_AL, "C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_weighted_unifrac_AL_22April2025.xlsx")

```

Next, we will build the models for CRP|AL and PC1|Sugar&Dairy for Beta diversity. As above, you'll have to run it in the console for the package to work correctly. Below is the code you can copy into the console to do that:

```{r}
library(readxl)

mds_bray <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_bray_22April2025.xlsx")
mds_jaccard <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_jaccard_22April2025.xlsx")
mds_unweighted_unifrac <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_unweighted_unifrac_22April2025.xlsx")
mds_weighted_unifrac <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_weighted_unifrac_22April2025.xlsx")

library(piecewiseSEM)
library(lme4)
library(semEff)

#Now the models: 

bray_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_bray),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_bray),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_bray, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_bray), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_bray),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_bray),
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = mds_bray),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = mds_bray),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month), data = mds_bray),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+ BMI + (1|Study_year) + (1|SampleID_Original), data = mds_bray),
  lmer(MDS3 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original) , data = mds_bray)
)


jaccard_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_jaccard),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_jaccard),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_jaccard, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_jaccard), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_jaccard),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_jaccard),
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = mds_jaccard),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = mds_jaccard),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month) + (1|AQ4_Sex), data = mds_jaccard),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) , data = mds_jaccard),
  lmer(MDS3 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|AQ4_Sex)+  BMI + (1|Study_year)  + (1|Abx_last_month), data = mds_jaccard)
)

unweighted_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_unweighted_unifrac),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_unweighted_unifrac),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_unweighted_unifrac, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_unweighted_unifrac), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_unweighted_unifrac),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_unweighted_unifrac),
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = mds_unweighted_unifrac),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = mds_unweighted_unifrac),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month) + (1|AQ4_Sex) + (1|Site), data = mds_unweighted_unifrac),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI  + (1|SampleID_Original) +  (1|AQ4_Sex), data = mds_unweighted_unifrac),
  lmer(MDS3 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI + (1|Study_year) + (1|SampleID_Original) + (1|AQ4_Sex), data = mds_unweighted_unifrac)
)

weighted_CRP_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_weighted_unifrac),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_weighted_unifrac),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_weighted_unifrac, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_weighted_unifrac), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_weighted_unifrac),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_weighted_unifrac),
  lmer(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI +  (1|SampleID_Original), data = mds_weighted_unifrac),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI, data = mds_weighted_unifrac),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI + (1|Study_year) + (1|SampleID_Original)  + (1|Site), data = mds_weighted_unifrac),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI + (1|Study_year) + (1|SampleID_Original) + (1|AQ4_Sex) + (1|Site), data = mds_weighted_unifrac))

#Now do CRP + sugar and dairy

bray_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_bray),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_bray),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_bray, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_bray), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_bray),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_bray),
  lmer(CRP ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + (1|AQ4_Sex)+ BMI , data = mds_bray),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI, data = mds_bray),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_bray),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + (1|smoke_household) + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site), data = mds_bray),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI + (1|Study_year) + (1|SampleID_Original), data = mds_bray),
  lmer(MDS3 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original), data = mds_bray)
)

jaccard_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_jaccard),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_jaccard),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_jaccard, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_jaccard), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_jaccard),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_jaccard),
  lmer(CRP ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI +  (1|AQ4_Sex), data = mds_jaccard),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI, data = mds_jaccard),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_jaccard),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI +  (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_jaccard),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI +  (1|Study_year) + (1|SampleID_Original), data = mds_jaccard),
  lmer(MDS3 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI +  (1|Study_year)  + (1|Site) + (1|AQ4_Sex), data = mds_jaccard)
)

unweighted_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_unweighted_unifrac),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_unweighted_unifrac),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_unweighted_unifrac, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_unweighted_unifrac), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_unweighted_unifrac),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_unweighted_unifrac),
  lmer(CRP ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI +  (1|AQ4_Sex), data = mds_unweighted_unifrac),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI, data = mds_unweighted_unifrac),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_unweighted_unifrac),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI +  (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_unweighted_unifrac),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI +   (1|SampleID_Original), data = mds_unweighted_unifrac),
  lmer(MDS3 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI +  (1|smoke_household) + (1|Study_year) + (1|SampleID_Original) + (1|AQ4_Sex) , data = mds_unweighted_unifrac)
)

weighted_CRP_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age + (1|AQ4_Sex) + (1|Site)  + (1|SampleID_Original), data = mds_weighted_unifrac),
 lmer(Income_range ~ Employment_limited + AQ3_Age +  (1|SampleID_Original), data = mds_weighted_unifrac),
glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_weighted_unifrac, family=binomial),
  lmer(food_insecurity2 ~ Income_range + AQ3_Age + (1|Site), data = mds_weighted_unifrac), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_weighted_unifrac),
  lmer(overcrowding_score ~ Income_range + AQ3_Age +  (1|Study_year), data = mds_weighted_unifrac),
  lmer(CRP ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI +  (1|AQ4_Sex), data = mds_weighted_unifrac),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI, data = mds_weighted_unifrac),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_weighted_unifrac),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI +   (1|Study_year) + (1|SampleID_Original) + (1|Site), data = mds_weighted_unifrac),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ CRP + overcrowding_score + AQ3_Age  + BMI +   (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex) , data = mds_weighted_unifrac)  )
```

Beta diversity models with AL:
```{r}
library(readxl)

mds_bray_AL <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_bray_AL_22April2025.xlsx")
mds_jaccard_AL <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_jaccard_AL_22April2025.xlsx")
mds_unweighted_unifrac_AL <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_unweighted_unifrac_AL_22April2025.xlsx")
mds_weighted_unifrac_AL <- read_xlsx("C:\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\mds_weighted_unifrac_AL_22April2025.xlsx")

mds_bray_AL$Education <- as.numeric(mds_bray_AL$Education)
mds_bray_AL$Employment_limited <- as.numeric(mds_bray_AL$Employment_limited)
mds_bray_AL$overcrowding_score <- as.numeric(mds_bray_AL$overcrowding_score)
mds_bray_AL$Own_home <- as.numeric(mds_bray_AL$Own_home)

mds_jaccard_AL$Education <- as.numeric(mds_jaccard_AL$Education)
mds_jaccard_AL$Employment_limited <- as.numeric(mds_jaccard_AL$Employment_limited)
mds_jaccard_AL$overcrowding_score <- as.numeric(mds_jaccard_AL$overcrowding_score)
mds_jaccard_AL$Own_home <- as.numeric(mds_jaccard_AL$Own_home)

mds_unweighted_unifrac_AL$Education <- as.numeric(mds_unweighted_unifrac_AL$Education)
mds_unweighted_unifrac_AL$Employment_limited <- as.numeric(mds_unweighted_unifrac_AL$Employment_limited)
mds_unweighted_unifrac_AL$overcrowding_score <- as.numeric(mds_unweighted_unifrac_AL$overcrowding_score)
mds_unweighted_unifrac_AL$Own_home <- as.numeric(mds_unweighted_unifrac_AL$Own_home)

mds_weighted_unifrac_AL$Education <- as.numeric(mds_weighted_unifrac_AL$Education)
mds_weighted_unifrac_AL$Employment_limited <- as.numeric(mds_weighted_unifrac_AL$Employment_limited)
mds_weighted_unifrac_AL$overcrowding_score <- as.numeric(mds_weighted_unifrac_AL$overcrowding_score)
mds_weighted_unifrac_AL$Own_home <- as.numeric(mds_weighted_unifrac_AL$Own_home)



bray_AL_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_bray_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_bray_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_bray_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_bray_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_bray_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_bray_AL),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_bray_AL),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = mds_bray_AL),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|smoke_household) + (1|Study_year) + (1|SampleID_Original) + (1|Site), data = mds_bray_AL),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age +  BMI  + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_bray_AL),
  lmer(MDS3 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI  + (1|Study_year) + (1|SampleID_Original) + (1|Site) , data = mds_bray_AL)
)



jaccard_AL_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_jaccard_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_jaccard_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_jaccard_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_jaccard_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_jaccard_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_jaccard_AL),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_jaccard_AL),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = mds_jaccard_AL),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|Abx_last_month), data = mds_jaccard_AL),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age +  BMI   + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|Abx_last_month) , data = mds_jaccard_AL),
  lmer(MDS3 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI   + (1|Study_year) + (1|Site) + (1|AQ4_Sex), data = mds_jaccard_AL)
)

unweighted_AL_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_unweighted_unifrac_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_unweighted_unifrac_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_unweighted_unifrac_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_unweighted_unifrac_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_unweighted_unifrac_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_unweighted_unifrac_AL),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_unweighted_unifrac_AL),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = mds_unweighted_unifrac_AL),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI  + (1|Study_year) + (1|Site) + (1|Abx_last_month) , data = mds_unweighted_unifrac_AL),
  lm(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age +  BMI, data = mds_unweighted_unifrac_AL),
  lmer(MDS3 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI   + (1|smoke_household) + (1|Study_year)  + (1|AQ4_Sex)+ (1|Abx_last_month), data = mds_unweighted_unifrac_AL)
)

weighted_AL_1 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_weighted_unifrac_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_weighted_unifrac_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_weighted_unifrac_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_weighted_unifrac_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_weighted_unifrac_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_weighted_unifrac_AL),
  lmer(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_weighted_unifrac_AL),
  lmer(PC1 ~ food_insecurity2 + AQ3_Age + (1|AQ4_Sex)+ (1|smoke_household) + BMI +  (1|SampleID_Original), data = mds_weighted_unifrac_AL),
  lmer(MDS1 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + (1|SampleID_Original) + (1|Site), data = mds_weighted_unifrac_AL),
  lmer(MDS2 ~ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age +  BMI  + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_weighted_unifrac_AL)
)

#Now do AL + sugar and dairy


bray_AL_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_bray_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_bray_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_bray_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_bray_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_bray_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_bray_AL),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_bray_AL),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = mds_bray_AL),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_bray_AL),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|smoke_household) + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|Abx_last_month) , data = mds_bray_AL),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_bray_AL),
  lmer(MDS3 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +  (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_bray_AL)
)

jaccard_AL_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_jaccard_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_jaccard_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_jaccard_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_jaccard_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_jaccard_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_jaccard_AL),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_jaccard_AL),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = mds_jaccard_AL),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_jaccard_AL),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|Abx_last_month), data = mds_jaccard_AL),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +  (1|Study_year) + (1|SampleID_Original) + (1|Site), data = mds_jaccard_AL),
  lmer(MDS3 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +  (1|Study_year) + (1|Site) + (1|AQ4_Sex), data = mds_jaccard_AL)
)


unweighted_AL_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_unweighted_unifrac_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_unweighted_unifrac_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_unweighted_unifrac_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_unweighted_unifrac_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_unweighted_unifrac_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_unweighted_unifrac_AL),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_unweighted_unifrac_AL),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = mds_unweighted_unifrac_AL),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_unweighted_unifrac_AL),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +  (1|Study_year)  + (1|Site) + (1|Abx_last_month), data = mds_unweighted_unifrac_AL),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +   (1|SampleID_Original), data = mds_unweighted_unifrac_AL),
  lmer(MDS3 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +  (1|smoke_household) + (1|Study_year) + (1|SampleID_Original) + (1|AQ4_Sex)+ (1|Abx_last_month), data = mds_unweighted_unifrac_AL)
)

weighted_AL_2 <- psem(
  lmer(Employment_limited ~ Education + AQ3_Age  + (1|Site)  + (1|SampleID_Original) , data = mds_weighted_unifrac_AL),
  lm(Income_range ~ Employment_limited + AQ3_Age, data = mds_weighted_unifrac_AL),
  glmer(healthcare_access ~ Employment_limited + AQ3_Age +  (1|Study_year) , data = mds_weighted_unifrac_AL, family=binomial),
  lm(food_insecurity2 ~ Income_range, data = mds_weighted_unifrac_AL), 
  lm(Own_home ~ Income_range + AQ3_Age, data = mds_weighted_unifrac_AL),
  lm(overcrowding_score ~ Income_range + AQ3_Age, data = mds_weighted_unifrac_AL),
  lmer(sum_risk ~ sugar + dairy + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age +  BMI +  (1|SampleID_Original)+ (1|Study_year), data = mds_weighted_unifrac_AL),
  lm(dairy ~ food_insecurity2 + AQ3_Age+ BMI , data = mds_weighted_unifrac_AL),
  lm(sugar ~ food_insecurity2 + AQ3_Age + BMI, data = mds_weighted_unifrac_AL),
  lmer(MDS1 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +   (1|SampleID_Original) + (1|Site), data = mds_weighted_unifrac_AL),
  lmer(MDS2 ~ sugar + dairy + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI +   (1|Study_year) + (1|SampleID_Original) + (1|Site) + (1|AQ4_Sex), data = mds_weighted_unifrac_AL)
)



```


##Refined beta diversity models 
```{r}
##CRP and PC1##
bray_unique_samples <- mds_bray %>% distinct(SampleID_Original, .keep_all = TRUE) 
jaccard_unique_samples <- mds_jaccard %>% distinct(SampleID_Original, .keep_all = TRUE)
unweighted_unique_samples <- mds_unweighted_unifrac %>% distinct(SampleID_Original, .keep_all = TRUE)
weighted_unique_samples <- mds_weighted_unifrac %>% distinct(SampleID_Original, .keep_all = TRUE)

bray_CRP_1 <- psem(
  lm(Employment_limited ~ Abx_last_month + Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = bray_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = bray_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = bray_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = bray_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = bray_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = bray_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = bray_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = bray_unique_samples),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = bray_unique_samples),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = bray_unique_samples),
  lm(MDS3 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site , data = bray_unique_samples)
)

bray_CRP_1_sem_effects <- semEff(bray_CRP_1, R = 1000, seed = 123)

jaccard_CRP_PC1 <- psem(
  lm(Employment_limited ~ Abx_last_month +Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = jaccard_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = jaccard_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = jaccard_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = jaccard_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = jaccard_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = jaccard_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = jaccard_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = jaccard_unique_samples),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = jaccard_unique_samples),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = jaccard_unique_samples),
  lm(MDS3 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site , data = jaccard_unique_samples)
)

jaccard_CRP_PC1_sem_effects <- semEff(jaccard_CRP_PC1, R = 1000, seed = 123)

unweighted_CRP_PC1 <- psem(
  lm(Employment_limited ~ Abx_last_month +Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = unweighted_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = unweighted_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = unweighted_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = unweighted_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = unweighted_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = unweighted_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = unweighted_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = unweighted_unique_samples),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = unweighted_unique_samples),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = unweighted_unique_samples),
  lm(MDS3 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site , data = unweighted_unique_samples))
  
 unweighted_CRP_PC1_sem_effects <- semEff(unweighted_CRP_PC1, R = 1000, seed = 123)

weighted_CRP_PC1 <- psem(
  lm(Employment_limited ~ Abx_last_month +Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = weighted_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = weighted_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = weighted_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = weighted_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = weighted_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = weighted_unique_samples),
  lm(CRP ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = weighted_unique_samples),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = weighted_unique_samples),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = weighted_unique_samples),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = weighted_unique_samples))
  
weighted_CRP_PC1_sem_effects <- semEff(weighted_CRP_PC1, R = 1000, seed = 123)

##CRP and sugar & dairy consumption##

bray_CRP_DS <- psem(
  lm(Employment_limited ~  Abx_last_month + Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = bray_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = bray_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = bray_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = bray_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = bray_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = bray_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = bray_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = bray_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = bray_unique_samples),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = bray_unique_samples),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = bray_unique_samples),
  lm(MDS3 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = bray_unique_samples))
  
bray_CRP_DS_sem_effects <- semEff(bray_CRP_DS, R = 1000, seed = 123)


jaccard_CRP_DS <- psem(
  lm(Employment_limited ~  Abx_last_month + Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = jaccard_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = jaccard_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = jaccard_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = jaccard_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = jaccard_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = jaccard_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = jaccard_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = jaccard_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = jaccard_unique_samples),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = jaccard_unique_samples),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = jaccard_unique_samples),
  lm(MDS3 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = jaccard_unique_samples))
  
jaccard_CRP_DS_sem_effects <- semEff(jaccard_CRP_DS, R = 1000, seed = 123)

unweighted_CRP_DS <- psem(
  lm(Employment_limited ~  Abx_last_month + Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = unweighted_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = unweighted_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = unweighted_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = unweighted_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = unweighted_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = unweighted_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = unweighted_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = unweighted_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = unweighted_unique_samples),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = unweighted_unique_samples),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = unweighted_unique_samples),
  lm(MDS3 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = unweighted_unique_samples))
  
unweighted_CRP_DS_sem_effects <- semEff(unweighted_CRP_DS, R = 1000, seed = 123)

weighted_CRP_DS <- psem(
  lm(Employment_limited ~  Abx_last_month + Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = weighted_unique_samples),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = weighted_unique_samples),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = weighted_unique_samples),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = weighted_unique_samples),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = weighted_unique_samples),
  lm(overcrowding_score ~ Income_range + AQ3_Age + Study_year, data = weighted_unique_samples),
  lm(CRP ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = weighted_unique_samples),
  lm(dairy ~ food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = weighted_unique_samples),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = weighted_unique_samples),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = weighted_unique_samples),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + CRP + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = weighted_unique_samples))
  

weighted_CRP_DS_sem_effects <- semEff(weighted_CRP_DS, R = 1000, seed = 123)
```

Refined beta diversity investigating AL 
```{r}
bray_unique_samples_AL <- mds_bray_AL %>% distinct(SampleID_Original, .keep_all = TRUE) 
jaccard_unique_samples_AL <- mds_jaccard_AL %>% distinct(SampleID_Original, .keep_all = TRUE)
unweighted_unique_samples_AL <- mds_unweighted_unifrac_AL %>% distinct(SampleID_Original, .keep_all = TRUE)
weighted_unique_samples_AL <- mds_weighted_unifrac_AL %>% distinct(SampleID_Original, .keep_all = TRUE)

bray_AL_1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = bray_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = bray_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = bray_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = bray_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = bray_unique_samples_AL),
  lm(overcrowding_score ~ BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = bray_unique_samples_AL),
  lm(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = bray_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = bray_unique_samples_AL),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = bray_unique_samples_AL),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = bray_unique_samples_AL),
  lm(MDS3 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site , data = bray_unique_samples_AL)
)

bray_AL_1_sem_effects <- semEff(bray_AL_1, R = 1000, seed = 123)

jaccard_AL_PC1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = jaccard_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = jaccard_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = jaccard_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = jaccard_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = jaccard_unique_samples_AL),
  lm(overcrowding_score ~ BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = jaccard_unique_samples_AL),
  lm(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = jaccard_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = jaccard_unique_samples_AL),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = jaccard_unique_samples_AL),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = jaccard_unique_samples_AL),
  lm(MDS3 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site , data = jaccard_unique_samples_AL)
)

jaccard_AL_PC1_sem_effects <- semEff(jaccard_AL_PC1, R = 1000, seed = 123)

unweighted_AL_PC1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = unweighted_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = unweighted_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = unweighted_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = unweighted_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = unweighted_unique_samples_AL),
  lm(overcrowding_score ~ BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = unweighted_unique_samples_AL),
  lm(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = unweighted_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = unweighted_unique_samples_AL),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = unweighted_unique_samples_AL),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = unweighted_unique_samples_AL),
  lm(MDS3 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site , data = unweighted_unique_samples_AL))
  
unweighted_AL_PC1_sem_effects <- semEff(unweighted_AL_PC1, R = 1000, seed = 123)

weighted_AL_PC1 <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = weighted_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = weighted_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = weighted_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = weighted_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = weighted_unique_samples_AL),
  lm(overcrowding_score ~ BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = weighted_unique_samples_AL),
  lm(sum_risk ~ PC1 + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + BMI + AQ4_Sex + smoke_household+  Site, data = weighted_unique_samples_AL),
  lm(PC1 ~ food_insecurity2 + Own_home + Income_range + AQ3_Age + BMI  + Study_year, data = weighted_unique_samples_AL),
  lm(MDS1 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = weighted_unique_samples_AL),
  lm(MDS2 ~ PC1 + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household + Abx_last_month+  Site, data = weighted_unique_samples_AL))
  
weighted_AL_PC1_sem_effects <- semEff(weighted_AL_PC1, R = 1000, seed = 123)

##CRP and sugar & dairy consumption##

bray_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = bray_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = bray_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = bray_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = bray_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = bray_unique_samples_AL),
  lm(overcrowding_score ~  BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = bray_unique_samples_AL),
  lm(sum_risk ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = bray_unique_samples_AL),
  lm(dairy ~ healthcare_access + food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = bray_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = bray_unique_samples_AL),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = bray_unique_samples_AL),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = bray_unique_samples_AL),
  lm(MDS3 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = bray_unique_samples_AL))
  
bray_AL_DS_sem_effects <- semEff(bray_AL_DS, R = 1000, seed = 123)


jaccard_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = jaccard_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = jaccard_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = jaccard_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = jaccard_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = jaccard_unique_samples_AL),
  lm(overcrowding_score ~  BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = jaccard_unique_samples_AL),
  lm(sum_risk ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = jaccard_unique_samples_AL),
  lm(dairy ~ healthcare_access + food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = jaccard_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = jaccard_unique_samples_AL),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = jaccard_unique_samples_AL),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = jaccard_unique_samples_AL),
  lm(MDS3 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = jaccard_unique_samples_AL))
  
jaccard_AL_DS_sem_effects <- semEff(jaccard_AL_DS, R = 1000, seed = 123)

unweighted_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = unweighted_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = unweighted_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = unweighted_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = unweighted_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = unweighted_unique_samples_AL),
  lm(overcrowding_score ~  BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = unweighted_unique_samples_AL),
  lm(sum_risk ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = unweighted_unique_samples_AL),
  lm(dairy ~ healthcare_access + food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = unweighted_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = unweighted_unique_samples_AL),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = unweighted_unique_samples_AL),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = unweighted_unique_samples_AL),
  lm(MDS3 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = unweighted_unique_samples_AL))
  
unweighted_AL_DS_sem_effects <- semEff(unweighted_AL_DS, R = 1000, seed = 123)

weighted_AL_DS <- psem(
  lm(Employment_limited ~ Education + BMI + AQ3_Age  +  AQ4_Sex  +  Site    , data = weighted_unique_samples_AL),
  lm(Income_range ~ Employment_limited + Education + AQ3_Age, data = weighted_unique_samples_AL),
  lm(healthcare_access ~ Employment_limited + Income_range + AQ3_Age + Study_year + smoke_household, data = weighted_unique_samples_AL),
  lm(food_insecurity2 ~ Income_range + AQ3_Age +  Site, data = weighted_unique_samples_AL),
  lm(Own_home ~ Income_range + Employment_limited + AQ3_Age + smoke_household, data = weighted_unique_samples_AL),
  lm(overcrowding_score ~  BMI +Employment_limited +Education +Income_range + AQ3_Age + Study_year, data = weighted_unique_samples_AL),
  lm(sum_risk ~ dairy + sugar + Own_home+ overcrowding_score+ healthcare_access + AQ3_Age + AQ4_Sex+ BMI + smoke_household+  Site, data = weighted_unique_samples_AL),
  lm(dairy ~ healthcare_access + food_insecurity2 + Income_range + overcrowding_score + Employment_limited + AQ3_Age + smoke_household + BMI, data = weighted_unique_samples_AL),
  lm(sugar ~ food_insecurity2 + Income_range + dairy + overcrowding_score + AQ3_Age + BMI, data = weighted_unique_samples_AL),
  lm(MDS1 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = weighted_unique_samples_AL),
  lm(MDS2 ~ dairy + sugar  + healthcare_access+ Income_range + Own_home + Education + overcrowding_score + AQ3_Age + BMI + sum_risk + Study_year +  AQ4_Sex + smoke_household+ Abx_last_month+  Site, data = weighted_unique_samples_AL))
  

weighted_AL_DS_sem_effects <- semEff(weighted_AL_DS, R = 1000, seed = 123)


```


Last thing is to check if "everything" model performs better than looking at a causal pathway
```{r}
#CRP_PC1
bray_CRP_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples))
    
jaccard_CRP_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples))


unweighted_CRP_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples))

weighted_CRP_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples))

#CRP_DS

bray_CRP_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples))

jaccard_CRP_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples))

unweighted_CRP_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples))

weighted_CRP_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ CRP + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples))


#AL_PC1

bray_AL_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples_AL),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples_AL))

jaccard_AL_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples_AL),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples_AL))


unweighted_AL_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples_AL),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples_AL))

weighted_AL_PC1_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples_AL))

#AL_DS

bray_AL_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month , data = bray_unique_samples_AL),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = bray_unique_samples_AL))

jaccard_AL_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples_AL),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = jaccard_unique_samples_AL))


unweighted_AL_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples_AL),
    lm(MDS3 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = unweighted_unique_samples_AL))

weighted_AL_DS_all <- psem(lm(MDS1 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples_AL), 
      lm(MDS2 ~ Employment_limited + Education + Income_range + Own_home + food_insecurity2+ dairy + sugar + healthcare_access+ sum_risk + overcrowding_score + AQ3_Age + BMI + Study_year + smoke_household + AQ4_Sex + Site+ Abx_last_month, data = weighted_unique_samples_AL))

```

##Phylogenetic Analyses
Using the results from alpha and beta diversity analyses, we will investigate which specific taxa are different by relevant SDoH variables. We will focus on income. 
We could also investigate quartiles of PC1 scores, education, and healthcare access? Although these were not consistent across beta diversity analyses. 

```{r}
#First get the data into R and make phyloseq objects from the taxonomy tables. 

library(tidyverse)
#library(phyloseq)
library(ANCOMBC)
library(DT)

phyla = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-phyla.tsv")
phyla_relabel = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-phyla-relab.tsv")
family_relabel = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-family-relab.tsv")
genus_relabel = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-genera-relab.tsv")

#Make quartiles for PC1
REACH_metadata_5$quartile <- with(REACH_metadata_5, cut(REACH_metadata_5$PC1, 
                                breaks=quantile(REACH_metadata_5$PC1, probs=seq(0,1, by=0.25), na.rm=TRUE), include.lowest=TRUE, labels=c("1", "2", "3", "4")))

genus_matrix = genus_relabel %>% column_to_rownames("SampleID") %>% as.matrix()
genus_phylo = otu_table(genus_matrix, taxa_are_rows = T)
meta_phylo = REACH_metadata_5 %>% column_to_rownames("Sample_ID") %>% 
  sample_data()

#genus_p = phyloseq(genus_phylo, meta_phylo)

family_matrix = family_relabel %>% column_to_rownames("SampleID") %>% as.matrix()
family_phylo = otu_table(family_matrix, taxa_are_rows = T)

family_p = phyloseq(family_phylo, meta_phylo)

phyla_matrix = phyla_relabel %>% column_to_rownames("SampleID") %>% as.matrix()
phyla_phylo = otu_table(phyla_matrix, taxa_are_rows = T)

#phyla_p = phyloseq(phyla_phylo, meta_phylo)

#Now we will run ANCOM-BC for income. 

set.seed(2911)
phyla_output = ancombc2(data= phyla_p, fix_formula = "Employment_limited + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access + CRP + overcrowding_score + AQ3_Age + BMI", group = "Income_range", global = TRUE, pairwise = TRUE, dunnet = TRUE)

#adding in random effects caused the log fold change and the w statistic to go to 0. We don't have the sample size to get it to work correctly. 

phyla_tab_sens = phyla_output$pseudo_sens_tab
phyla_tab_sens %>%
  datatable(caption = "Sensitivity Scores") %>% 
  formatRound(colnames(phyla_tab_sens)[-1], digits = 2)

res_phyla = phyla_output$res

family_output = ancombc2(data= family_p, fix_formula = "Employment_limited + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access + CRP + overcrowding_score + AQ3_Age + BMI", group = "Income_range", global = TRUE, pairwise = TRUE, dunnet = TRUE)

family_tab_sens = family_output$pseudo_sens_tab
family_tab_sens %>%
  datatable(caption = "Sensitivity Scores") %>% 
  formatRound(colnames(family_tab_sens)[-1], digits = 2)

res_family = family_output$res

genus_output = ancombc2(data= genus_p, fix_formula = "Employment_limited + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access + CRP + overcrowding_score + AQ3_Age + BMI", group = "Income_range", global = TRUE, pairwise = TRUE, dunnet = TRUE)

genus_tab_sens = genus_output$pseudo_sens_tab
genus_tab_sens %>%
  datatable(caption = "Sensitivity Scores") %>% 
  formatRound(colnames(genus_tab_sens)[-1], digits = 2)

res_genus = genus_output$res

#Now grouping by healthcare access
healthcare_genus_output = ancombc2(data= genus_p, fix_formula = "Employment_limited + Income_range + Own_home + food_insecurity2+ PC1 + healthcare_access + CRP + overcrowding_score + AQ3_Age + BMI", group = "healthcare_access", global = TRUE, pairwise = TRUE, dunnet = TRUE)

genus_tab_sens_healthcare = healthcare_genus_output$pseudo_sens_tab
genus_tab_sens_healthcare %>%
  datatable(caption = "Sensitivity Scores") %>% 
  formatRound(colnames(genus_tab_sens_healthcare)[-1], digits = 2)

res_genus_healthcare = healthcare_genus_output$res

#Now for education 
education_genus_output = ancombc2(data= genus_p, fix_formula = "PC1 + Income_range + healthcare_access+ CRP + Education + overcrowding_score + AQ3_Age + BMI", group = "Education", global = TRUE, pairwise = TRUE, dunnet = TRUE)

genus_tab_sens_education = education_genus_output$pseudo_sens_tab
genus_tab_sens_education %>%
  datatable(caption = "Sensitivity Scores") %>% 
  formatRound(colnames(genus_tab_sens_education)[-1], digits = 2)

res_genus_education = education_genus_output$res

##Next make PC1 quantiles and run that!

diet_genus_output = ancombc2(data= genus_p, fix_formula = "Employment_limited + Income_range + Own_home + food_insecurity2+ quartile + healthcare_access + CRP + overcrowding_score + AQ3_Age + BMI", group = "quartile", global = TRUE, pairwise = TRUE, dunnet = TRUE)

genus_tab_sens_diet = diet_genus_output$pseudo_sens_tab
genus_tab_sens_diet %>%
  datatable(caption = "Sensitivity Scores") %>% 
  formatRound(colnames(genus_tab_sens_diet)[-1], digits = 2)

res_genus_diet = diet_genus_output$res

#The q values are 1 for the genus level because the sample size is too small for so many tests. Switching to glmm with negative binomial distribution for better analysis. 
```

Turns out the glmm doesn't show anything either? This might be a problem with the way I've written the code. 
```{r}
library(glmmTMB)
library(multcomp)
library(car)
library(tidyverse)
library(fdrtool)

phyla_transposed <- as.data.frame(t(phyla_matrix))
meta_phylo_df = REACH_metadata_5 %>% column_to_rownames("Sample_ID") 
phyla_glmm = merge(phyla_transposed, meta_phylo_df, by = "row.names")
#add all the phyla as columns 

Income_range_estimatematrix1 = mat.or.vec(31,2)
Income_range_pvaluematrix1 = mat.or.vec(31,2)
PC1_estimatematrix1 = mat.or.vec(31,2)
PC1_pvaluematrix1 = mat.or.vec(31,2)
healthcare_access_estimatematrix1 = mat.or.vec(31,2)
healthcare_access_pvaluematrix1 = mat.or.vec(31,2)


for(i in 2:32) {
  variable = phyla_glmm[,i]
  b = try(glmmTMB(variable ~ PC1 + Income_range + healthcare_access+ CRP + Education + overcrowding_score + AQ3_Age + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month) + (1|AQ4_Sex) + (1|Site), 
                  data = phyla_glmm, family = nbinom2, na.action = na.omit, control = control))
  if("try-error" %in% class(b)) anova <- blank
  else anova = Anova(b)
  Income_range_estimatematrix1[i-1,2] = anova[2,1]
  Income_range_pvaluematrix1[i-1,2] = anova[2,3]
  PC1_estimatematrix1[i-1,2] = anova[1,1]
  PC1_pvaluematrix1[i-1,2] = anova[1,3]
  healthcare_access_estimatematrix1[i-1,2] = anova[3,1]
  healthcare_access_pvaluematrix1[i-1,2] = anova[3,3]
  
  Income_range_estimatematrix1[i-1,1] = names(phyla_glmm)[i]
  Income_range_pvaluematrix1[i-1,1] = names(phyla_glmm)[i]
  PC1_estimatematrix1[i-1,1] = names(phyla_glmm)[i]
  PC1_pvaluematrix1[i-1,1] = names(phyla_glmm)[i]
  healthcare_access_estimatematrix1[i-1,1] = names(phyla_glmm)[i]
  healthcare_access_pvaluematrix1[i-1,1] = names(phyla_glmm)[i]}


phyla_income = bind_cols(as.data.frame(Income_range_estimatematrix1[,1:2]), 
                         as.data.frame(Income_range_pvaluematrix1[,2]))
write.csv(phyla_income, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\phyla_income.csv")
phyla_income_nona = filter(phyla_income, phyla_income[,3] != "NaN")
phyla_income_nona[,3] = as.numeric(as.character(phyla_income_nona[,3]))
phyla_income_corrected = bind_cols(phyla_income_nona, 
                                   as.data.frame(fdrtool(phyla_income_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(phyla_income_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\phyla_income_corrected.csv")

phyla_PC1 = bind_cols(as.data.frame(PC1_estimatematrix1[,1:2]), 
                         as.data.frame(PC1_pvaluematrix1[,2]))
write.csv(phyla_PC1, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\phyla_PC1.csv")
phyla_PC1_nona = filter(phyla_PC1, phyla_PC1[,3] != "NaN")
phyla_PC1_nona[,3] = as.numeric(as.character(phyla_PC1_nona[,3]))
phyla_PC1_corrected = bind_cols(phyla_PC1_nona, 
                                   as.data.frame(fdrtool(phyla_PC1_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(phyla_PC1_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\phyla_PC1_corrected.csv")

phyla_healthcare = bind_cols(as.data.frame(healthcare_access_estimatematrix1[,1:2]), 
                         as.data.frame(healthcare_access_pvaluematrix1[,2]))
write.csv(phyla_healthcare, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\phyla_healthcare.csv")
phyla_healthcare_nona = filter(phyla_healthcare, phyla_healthcare[,3] != "NaN")
phyla_healthcare_nona[,3] = as.numeric(as.character(phyla_healthcare_nona[,3]))
phyla_healthcare_corrected = bind_cols(phyla_healthcare_nona, 
                                   as.data.frame(fdrtool(phyla_healthcare_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(phyla_healthcare_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\phyla_healthcare_corrected.csv")

##Now to do family level
family_relabel = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-family-relabel.tsv")

family_matrix = family_relabel %>% column_to_rownames("SampleID") %>% as.matrix()
family_transposed <- as.data.frame(t(family_matrix))
#meta_phylo_df = REACH_metadata_5 %>% column_to_rownames("Sample_ID") 
family_glmm = merge(family_transposed, meta_phylo_df, by = "row.names")


Income_range_estimatematrix2 = mat.or.vec(194,2)
Income_range_pvaluematrix2 = mat.or.vec(194,2)
PC1_estimatematrix2 = mat.or.vec(194,2)
PC1_pvaluematrix2 = mat.or.vec(194,2)
healthcare_access_estimatematrix2 = mat.or.vec(194,2)
healthcare_access_pvaluematrix2 = mat.or.vec(194,2)

for(i in 2:195) {
  variable = family_glmm[,i]
  b = try(glmmTMB(variable ~ PC1 + Income_range + healthcare_access+ CRP + Education + overcrowding_score + AQ3_Age + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month) + (1|AQ4_Sex) + (1|Site), 
                  data = family_glmm, family = nbinom2, na.action = na.omit, control = control))
  if("try-error" %in% class(b)) anova <- blank
  else anova = Anova(b)
  Income_range_estimatematrix2[i-1,2] = anova[2,1]
  Income_range_pvaluematrix2[i-1,2] = anova[2,3]
  PC1_estimatematrix2[i-1,2] = anova[1,1]
  PC1_pvaluematrix2[i-1,2] = anova[1,3]
  healthcare_access_estimatematrix2[i-1,2] = anova[3,1]
  healthcare_access_pvaluematrix2[i-1,2] = anova[3,3]
  
  Income_range_estimatematrix2[i-1,1] = names(family_glmm)[i]
  Income_range_pvaluematrix2[i-1,1] = names(family_glmm)[i]
  PC1_estimatematrix2[i-1,1] = names(family_glmm)[i]
  PC1_pvaluematrix2[i-1,1] = names(family_glmm)[i]
  healthcare_access_estimatematrix2[i-1,1] = names(family_glmm)[i]
  healthcare_access_pvaluematrix2[i-1,1] = names(family_glmm)[i]}


family_income = bind_cols(as.data.frame(Income_range_estimatematrix2[,1:2]), 
                         as.data.frame(Income_range_pvaluematrix2[,2]))
write.csv(family_income, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\family_income.csv")
family_income_nona = filter(family_income, family_income[,3] != "NaN")
family_income_nona[,3] = as.numeric(as.character(family_income_nona[,3]))
family_income_corrected = bind_cols(family_income_nona, 
                                   as.data.frame(fdrtool(family_income_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(family_income_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\family_income_corrected.csv")

family_PC1 = bind_cols(as.data.frame(PC1_estimatematrix2[,1:2]), 
                         as.data.frame(PC1_pvaluematrix2[,2]))
write.csv(family_PC1, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\family_PC1.csv")
family_PC1_nona = filter(family_PC1, family_PC1[,3] != "NaN")
family_PC1_nona[,3] = as.numeric(as.character(family_PC1_nona[,3]))
family_PC1_corrected = bind_cols(family_PC1_nona, 
                                   as.data.frame(fdrtool(family_PC1_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(family_PC1_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\family_PC1_corrected.csv")

family_healthcare = bind_cols(as.data.frame(healthcare_access_estimatematrix2[,1:2]), 
                         as.data.frame(healthcare_access_pvaluematrix2[,2]))
write.csv(family_healthcare, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\family_healthcare.csv")
family_healthcare_nona = filter(family_healthcare, family_healthcare[,3] != "NaN")
family_healthcare_nona[,3] = as.numeric(as.character(family_healthcare_nona[,3]))
family_healthcare_corrected = bind_cols(family_healthcare_nona, 
                                   as.data.frame(fdrtool(family_healthcare_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(family_healthcare_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\family_healthcare_corrected.csv")

#Lastly, do genus level

genus_matrix = genus_relabel %>% column_to_rownames("SampleID") %>% as.matrix()
genus_transposed <- as.data.frame(t(genus_matrix))
#meta_phylo_df = REACH_metadata_5 %>% column_to_rownames("Sample_ID") 
genus_glmm = merge(genus_transposed, meta_phylo_df, by = "row.names")


Income_range_estimatematrix3 = mat.or.vec(581,2)
Income_range_pvaluematrix3 = mat.or.vec(581,2)
PC1_estimatematrix3 = mat.or.vec(581,2)
PC1_pvaluematrix3 = mat.or.vec(581,2)
healthcare_access_estimatematrix3 = mat.or.vec(581,2)
healthcare_access_pvaluematrix3 = mat.or.vec(581,2)

for(i in 2:582) {
  variable = genus_glmm[,i]
  b = try(glmmTMB(variable ~ PC1 + Income_range + healthcare_access+ CRP + Education + overcrowding_score + AQ3_Age + (1|Study_year) + (1|SampleID_Original) + (1|Abx_last_month) + (1|AQ4_Sex) + (1|Site), 
                  data = genus_glmm, family = nbinom2, na.action = na.omit, control = control))
  if("try-error" %in% class(b)) anova <- blank
  else anova = Anova(b)
  Income_range_estimatematrix3[i-1,2] = anova[2,1]
  Income_range_pvaluematrix3[i-1,2] = anova[2,3]
  PC1_estimatematrix3[i-1,2] = anova[1,1]
  PC1_pvaluematrix3[i-1,2] = anova[1,3]
  healthcare_access_estimatematrix3[i-1,2] = anova[3,1]
  healthcare_access_pvaluematrix3[i-1,2] = anova[3,3]
  
  Income_range_estimatematrix3[i-1,1] = names(genus_glmm)[i]
  Income_range_pvaluematrix3[i-1,1] = names(genus_glmm)[i]
  PC1_estimatematrix3[i-1,1] = names(genus_glmm)[i]
  PC1_pvaluematrix3[i-1,1] = names(genus_glmm)[i]
  healthcare_access_estimatematrix3[i-1,1] = names(genus_glmm)[i]
  healthcare_access_pvaluematrix3[i-1,1] = names(genus_glmm)[i]}


genus_income = bind_cols(as.data.frame(Income_range_estimatematrix3[,1:2]), 
                         as.data.frame(Income_range_pvaluematrix3[,2]))
write.csv(genus_income, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\genus_income.csv")
genus_income_nona = filter(genus_income, genus_income[,3] != "NaN")
genus_income_nona[,3] = as.numeric(as.character(genus_income_nona[,3]))
genus_income_corrected = bind_cols(genus_income_nona, 
                                   as.data.frame(fdrtool(genus_income_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(genus_income_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\genus_income_corrected.csv")

genus_PC1 = bind_cols(as.data.frame(PC1_estimatematrix3[,1:2]), 
                         as.data.frame(PC1_pvaluematrix3[,2]))
write.csv(genus_PC1, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\genus_PC1.csv")
genus_PC1_nona = filter(genus_PC1, genus_PC1[,3] != "NaN")
genus_PC1_nona[,3] = as.numeric(as.character(genus_PC1_nona[,3]))
genus_PC1_corrected = bind_cols(genus_PC1_nona, 
                                   as.data.frame(fdrtool(genus_PC1_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(genus_PC1_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\genus_PC1_corrected.csv")

genus_healthcare = bind_cols(as.data.frame(healthcare_access_estimatematrix3[,1:2]), 
                         as.data.frame(healthcare_access_pvaluematrix3[,2]))
write.csv(genus_healthcare, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\genus_healthcare.csv")
genus_healthcare_nona = filter(genus_healthcare, genus_healthcare[,3] != "NaN")
genus_healthcare_nona[,3] = as.numeric(as.character(genus_healthcare_nona[,3]))
genus_healthcare_corrected = bind_cols(genus_healthcare_nona, 
                                   as.data.frame(fdrtool(genus_healthcare_nona[,3], 
                                                         statistic = "pvalue", plot = F)))
write.csv(genus_healthcare_corrected, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\genus_healthcare_corrected.csv")

```

Well, neither of those indicated any significant taxa. We will use a less conservative approach through ALDEx2 as a more exploratory method given our small sample size. 

```{r}
#if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")


#BiocManager::install("ALDEx2")

library(ALDEx2)


#vector of conditions must be in same order as the phyla dataframe

#phyla table must be the same length as the variables 
phyla_short <- merge(meta_phylo_df, phyla_transposed, by = "row.names")
phyla_short <- phyla_short[,146:176]
t_phyla_short <- as.data.frame(t(phyla_short))

phyla_testing_matrix = phyla %>% column_to_rownames("SampleID") %>% as.matrix()
phyla_testing_transposed <- as.data.frame(t(phyla_testing_matrix))
#meta_phylo_df = REACH_metadata_5 %>% column_to_rownames("Sample_ID") 
phyla_short_integer <- merge(meta_phylo_df, phyla_testing_transposed, by = "row.names")
phyla_short_integer <- phyla_short_integer[,146:176]
t_phyla_short_integer <- as.data.frame(t(phyla_short_integer))


mm_phyla <- model.matrix(~ as.factor(Income_range) + PC1 + as.factor(healthcare_access), meta_phylo_df)
clr_phyla <- aldex.clr(t_phyla_short_integer, mm_phyla, mc.samples = 128, denom = "all", verbose = F)
glm.phyla <- aldex.glm(clr_phyla, mm_phyla)
#Firmicutes D different for highest income group 

write.csv(glm.phyla, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\glm_phyla.csv")

#Now do family
family = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-family.tsv")
family_testing_matrix = family %>% column_to_rownames("SampleID") %>% as.matrix()
family_testing_transposed <- as.data.frame(t(family_testing_matrix))
#meta_phylo_df = REACH_metadata_5 %>% column_to_rownames("Sample_ID") 
family_short_integer <- merge(meta_phylo_df, family_testing_transposed, by = "row.names")
family_short_integer <- family_short_integer[,146:339]
t_family_short_integer <- as.data.frame(t(family_short_integer))

#mm_family <- model.matrix(~ as.factor(Income_range) + PC1 + as.factor(healthcare_access) + CRP + as.factor(Education) + overcrowding_score + AQ3_Age, meta_phylo_df)
clr_family <- aldex.clr(t_family_short_integer, mm_phyla, mc.samples = 128, denom = "all", verbose = F)
glm.family <- aldex.glm(clr_family, mm_phyla)

write.csv(glm.family, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\glm_family.csv")

#now we will do genus
genus = read_tsv("\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\feature-table-genera.tsv")
genus_testing_matrix = genus %>% column_to_rownames("SampleID") %>% as.matrix()
genus_testing_transposed <- as.data.frame(t(genus_testing_matrix))
#meta_phylo_df = REACH_metadata_5 %>% column_to_rownames("Sample_ID") 
genus_short_integer <- merge(meta_phylo_df, genus_testing_transposed, by = "row.names")
genus_short_integer <- genus_short_integer[,146:726]
t_genus_short_integer <- as.data.frame(t(genus_short_integer))

#mm_genus <- model.matrix(~ as.factor(Income_range) + PC1 + as.factor(healthcare_access) + CRP + as.factor(Education) + overcrowding_score + AQ3_Age, meta_phylo_df)
clr_genus <- aldex.clr(t_genus_short_integer, mm_phyla, mc.samples = 128, denom = "all", verbose = F)
glm.genus <- aldex.glm(clr_genus, mm_phyla)

write.csv(glm.genus, "\\Users\\carly\\OneDrive\\Desktop\\Publications\\REACH Microbiome Comparison\\SDOH paper\\glm_genus.csv")

```



